# Assemble genomes of model organisms by ANCHR

[TOC levels=1-3]: # " "
- [Assemble genomes of model organisms by ANCHR](#assemble-genomes-of-model-organisms-by-anchr)
- [*Saccharomyces cerevisiae* S288c](#saccharomyces-cerevisiae-s288c)
    - [Scer: download](#scer-download)
    - [Scer: trim](#scer-trim)
    - [Scer: down sampling](#scer-down-sampling)
    - [Scer: generate super-reads](#scer-generate-super-reads)
    - [Scer: create anchors](#scer-create-anchors)
    - [Scer: results](#scer-results)
    - [Scer: quality assessment](#scer-quality-assessment)
    - [Scer: anchor-long](#scer-anchor-long)
- [*Drosophila melanogaster* iso-1](#drosophila-melanogaster-iso-1)
    - [Dmel: download](#dmel-download)
    - [Dmel: trim](#dmel-trim)
    - [Dmel: down sampling](#dmel-down-sampling)
    - [Dmel: generate super-reads](#dmel-generate-super-reads)
    - [Dmel: create anchors](#dmel-create-anchors)
    - [Dmel: results](#dmel-results)
    - [Dmel: quality assessment](#dmel-quality-assessment)
- [*Caenorhabditis elegans* N2](#caenorhabditis-elegans-n2)
    - [Cele: download](#cele-download)
    - [Cele: trim](#cele-trim)
    - [Cele: down sampling](#cele-down-sampling)
    - [Cele: generate super-reads](#cele-generate-super-reads)
    - [Cele: create anchors](#cele-create-anchors)
    - [Cele: results](#cele-results)
    - [Cele: quality assessment](#cele-quality-assessment)
- [*Arabidopsis thaliana* Col-0](#arabidopsis-thaliana-col-0)
    - [Atha: download](#atha-download)
    - [Atha: trim](#atha-trim)
    - [Atha: down sampling](#atha-down-sampling)
    - [Atha: generate super-reads](#atha-generate-super-reads)
    - [Atha: create anchors](#atha-create-anchors)
    - [Atha: results](#atha-results)
    - [Atha: quality assessment](#atha-quality-assessment)

# *Saccharomyces cerevisiae* S288c

* Genome: [Ensembl 82](http://sep2015.archive.ensembl.org/Saccharomyces_cerevisiae/Info/Index)
* Proportion of paralogs (> 1000 bp): 0.058

## Scer: download

* Reference genome

```bash
mkdir -p ~/data/anchr/s288c/1_genome
cd ~/data/anchr/s288c/1_genome
wget -N ftp://ftp.ensembl.org/pub/release-82/fasta/saccharomyces_cerevisiae/dna/Saccharomyces_cerevisiae.R64-1-1.dna_sm.toplevel.fa.gz
faops order Saccharomyces_cerevisiae.R64-1-1.dna_sm.toplevel.fa.gz \
    <(for chr in {I,II,III,IV,V,VI,VII,VIII,IX,X,XI,XII,XIII,XIV,XV,XVI,Mito}; do echo $chr; done) \
    genome.fa
```

* Illumina

    ENA hasn't synced with SRA for PRJNA340312, download with prefetch from sratoolkit.

```bash
mkdir -p ~/data/anchr/s288c/2_illumina
cd ~/data/anchr/s288c/2_illumina
prefetch --progress 0.5 SRR4074255
fastq-dump --split-files SRR4074255  
find . -name "*.fastq" | parallel -j 2 pigz -p 8

ln -s SRR4074255_1.fastq.gz R1.fq.gz
ln -s SRR4074255_2.fastq.gz R2.fq.gz
```

* PacBio

    PacBio provides a dataset of *S. cerevisiae* strain
    [W303](https://github.com/PacificBiosciences/DevNet/wiki/Saccharomyces-cerevisiae-W303-Assembly-Contigs),
    while the reference strain S288c is not provided. So we use the dataset from
    [project PRJEB7245](https://www.ncbi.nlm.nih.gov/bioproject/PRJEB7245),
    [study ERP006949](https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=ERP006949), and
    [sample SAMEA4461733](https://www.ncbi.nlm.nih.gov/biosample/5850878). This is gathered with RS
    II and P6C4.

```bash
mkdir -p ~/data/anchr/s288c/3_pacbio
cd ~/data/anchr/s288c/3_pacbio

# download from sra
cat <<EOF > hdf5.txt
http://sra-download.ncbi.nlm.nih.gov/srapub_files/ERR1655118_ERR1655118_hdf5.tgz
http://sra-download.ncbi.nlm.nih.gov/srapub_files/ERR1655120_ERR1655120_hdf5.tgz
http://sra-download.ncbi.nlm.nih.gov/srapub_files/ERR1655122_ERR1655122_hdf5.tgz
http://sra-download.ncbi.nlm.nih.gov/srapub_files/ERR1655124_ERR1655124_hdf5.tgz

EOF
aria2c -x 9 -s 3 -c -i hdf5.txt

# untar
mkdir -p ~/data/anchr/s288c/3_pacbio/untar
cd ~/data/anchr/s288c/3_pacbio
tar xvfz ERR1655118_ERR1655118_hdf5.tgz --directory untar
#tar xvfz ERR1655120_ERR1655120_hdf5.tgz --directory untar
#tar xvfz ERR1655122_ERR1655122_hdf5.tgz --directory untar
#tar xvfz ERR1655124_ERR1655124_hdf5.tgz --directory untar

# convert .bax.h5 to .subreads.bam
mkdir -p ~/data/anchr/s288c/3_pacbio/bam
cd ~/data/anchr/s288c/3_pacbio/bam

source ~/share/pitchfork/deployment/setup-env.sh
for movie in m150412 m150415 m150417 m150421;
do 
    bax2bam ~/data/anchr/s288c/3_pacbio/untar/${movie}*.bax.h5
done

# convert .subreads.bam to fasta
mkdir -p ~/data/anchr/s288c/3_pacbio/fasta

for movie in m150412 m150415 m150417 m150421;
do
    if [ ! -e ~/data/anchr/s288c/3_pacbio/bam/${movie}*.subreads.bam ]; then
        continue
    fi

    samtools fasta \
        ~/data/anchr/s288c/3_pacbio/bam/${movie}*.subreads.bam \
        > ~/data/anchr/s288c/3_pacbio/fasta/${movie}.fasta
done

cd ~/data/anchr/s288c/3_pacbio
ln -s fasta/m150412.fasta pacbio.fasta
```

## Scer: combinations of different quality values and read lengths

* qual: 20, 25, and 30
* len: 120, 130, 140 and 150

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

# get the default adapter file
# anchr trim --help
scythe \
    2_illumina/R1.fq.gz \
    -q sanger \
    -M 100 \
    -a /home/wangq/.plenv/versions/5.18.4/lib/perl5/site_perl/5.18.4/auto/share/dist/App-Anchr/illumina_adapters.fa \
    --quiet \
    | pigz -p 4 -c \
    > 2_illumina/R1.scythe.fq.gz

scythe \
    2_illumina/R2.fq.gz \
    -q sanger \
    -M 100 \
    -a /home/wangq/.plenv/versions/5.18.4/lib/perl5/site_perl/5.18.4/auto/share/dist/App-Anchr/illumina_adapters.fa \
    --quiet \
    | pigz -p 4 -c \
    > 2_illumina/R2.scythe.fq.gz

cd ${BASE_DIR}
parallel --no-run-if-empty -j 6 "
        mkdir -p 2_illumina/Q{1}L{2}
        cd 2_illumina/Q{1}L{2}
        
        anchr trim \
            --noscythe \
            -q {1} -l {2} \
            ../R1.scythe.fq.gz ../R2.scythe.fq.gz \
            -o stdout \
            | bash
    " ::: 20 25 30 ::: 120 130 140 150

```

* Stats

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

printf "| %s | %s | %s | %s |\n" \
    "Name" "N50" "Sum" "#" \
    > stat.md
printf "|:--|--:|--:|--:|\n" >> stat.md

printf "| %s | %s | %s | %s |\n" \
    $(echo "Genome";   faops n50 -H -S -C 1_genome/genome.fa;) >> stat.md
printf "| %s | %s | %s | %s |\n" \
    $(echo "Illumina"; faops n50 -H -S -C 2_illumina/R1.fq.gz 2_illumina/R2.fq.gz;) >> stat.md
printf "| %s | %s | %s | %s |\n" \
    $(echo "PacBio";   faops n50 -H -S -C 3_pacbio/pacbio.fasta;) >> stat.md
printf "| %s | %s | %s | %s |\n" \
    $(echo "scythe";   faops n50 -H -S -C 2_illumina/R1.scythe.fq.gz 2_illumina/R2.scythe.fq.gz;) >> stat.md

for qual in 20 25 30; do
    for len in 120 130 140 150; do
        DIR_COUNT="${BASE_DIR}/2_illumina/Q${qual}L${len}"

        printf "| %s | %s | %s | %s |\n" \
            $(echo "Q${qual}L${len}"; faops n50 -H -S -C ${DIR_COUNT}/R1.fq.gz  ${DIR_COUNT}/R2.fq.gz;) \
            >> stat.md
    done
done

cat stat.md
```

| Name     |    N50 |        Sum |        # |
|:---------|-------:|-----------:|---------:|
| Genome   | 924431 |   12157105 |       17 |
| Illumina |    151 | 2939081214 | 19464114 |
| PacBio   |   8412 |  820962526 |   177100 |
| scythe   |    151 | 2856064236 | 19464114 |
| Q20L120  |    151 | 2671314209 | 17768540 |
| Q20L130  |    151 | 2637766736 | 17510028 |
| Q20L140  |    151 | 2589334780 | 17160084 |
| Q20L150  |    151 | 2540801666 | 16827778 |
| Q25L120  |    151 | 2468125943 | 16421630 |
| Q25L130  |    151 | 2431389599 | 16140530 |
| Q25L140  |    151 | 2383125715 | 15792300 |
| Q25L150  |    151 | 2348269619 | 15552026 |
| Q30L120  |    151 | 2244788901 | 14947056 |
| Q30L130  |    151 | 2203023127 | 14630350 |
| Q30L140  |    151 | 2147806869 | 14234194 |
| Q30L150  |    151 | 2104817905 | 13939512 |

## Scer: down sampling

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

ARRAY=( "2_illumina:original:8000000"
        "2_illumina/Q20L120:Q20L120:8000000"
        "2_illumina/Q20L130:Q20L130:8000000"
        "2_illumina/Q20L140:Q20L140:8000000"
        "2_illumina/Q20L150:Q20L150:8000000"
        "2_illumina/Q25L120:Q25L120:8000000"
        "2_illumina/Q25L130:Q25L130:8000000"
        "2_illumina/Q25L140:Q25L140:7000000"
        "2_illumina/Q25L150:Q25L150:7000000"
        "2_illumina/Q30L120:Q30L120:7000000"
        "2_illumina/Q30L130:Q30L130:7000000"
        "2_illumina/Q30L140:Q30L140:7000000"
        "2_illumina/Q30L150:Q30L150:7000000")

for group in "${ARRAY[@]}" ; do
    
    GROUP_DIR=$(group=${group} perl -e '@p = split q{:}, $ENV{group}; print $p[0];')
    GROUP_ID=$( group=${group} perl -e '@p = split q{:}, $ENV{group}; print $p[1];')
    GROUP_MAX=$(group=${group} perl -e '@p = split q{:}, $ENV{group}; print $p[2];')
    printf "==> %s \t %s \t %s\n" "$GROUP_DIR" "$GROUP_ID" "$GROUP_MAX"

    for count in $(perl -e 'print 1000000 * $_, q{ } for 1 .. 8');
    do
        if [[ "$count" -gt "$GROUP_MAX" ]]; then
            continue;
        fi
        
        echo "==> Group ${GROUP_ID}_${count}"
        DIR_COUNT="${BASE_DIR}/${GROUP_ID}_${count}"
        mkdir -p ${DIR_COUNT}
        
        if [ -e ${DIR_COUNT}/R1.fq.gz ]; then
            continue;
        fi
        
        seqtk sample -s${count} \
            ${BASE_DIR}/${GROUP_DIR}/R1.fq.gz ${count} \
            | pigz > ${DIR_COUNT}/R1.fq.gz
        seqtk sample -s${count} \
            ${BASE_DIR}/${GROUP_DIR}/R2.fq.gz ${count} \
            | pigz > ${DIR_COUNT}/R2.fq.gz
    done
done
```

## Scer: generate super-reads

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

perl -e '
    for my $n (
        qw{
        original
        Q20L120 Q20L130 Q20L140 Q20L150
        Q25L120 Q25L130 Q25L140 Q25L150
        Q30L120 Q30L130 Q30L140 Q30L150
        }
        )
    {
        for my $i ( 1 .. 8 ) {
            printf qq{%s_%d\n}, $n, ( 1000000 * $i );
        }
    }
    ' \
    | parallel --no-run-if-empty -j 6 "
        echo '==> Group {}'
        
        if [ ! -d ${BASE_DIR}/{} ]; then
            echo '    directory not exists'
            exit;
        fi        

        if [ -e ${BASE_DIR}/{}/pe.cor.fa ]; then
            echo '    pe.cor.fa already presents'
            exit;
        fi

        cd ${BASE_DIR}/{}
        anchr superreads \
            R1.fq.gz R2.fq.gz \
            --nosr -p 8 \
            -o superreads.sh
        bash superreads.sh
    "
```

Clear intermediate files.

```bash
cd $HOME/data/anchr/s288c

find . -type f -name "quorum_mer_db.jf"          | xargs rm
find . -type f -name "k_u_hash_0"                | xargs rm
find . -type f -name "readPositionsInSuperReads" | xargs rm
find . -type f -name "*.tmp"                     | xargs rm
find . -type f -name "pe.renamed.fastq"          | xargs rm
find . -type f -name "pe.cor.sub.fa"             | xargs rm
```

## Scer: create anchors

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

perl -e '
    for my $n (
        qw{
        original
        Q20L120 Q20L130 Q20L140 Q20L150
        Q25L120 Q25L130 Q25L140 Q25L150
        Q30L120 Q30L130 Q30L140 Q30L150
        }
        )
    {
        for my $i ( 1 .. 8 ) {
            printf qq{%s_%d\n}, $n, ( 1000000 * $i );
        }
    }
    ' \
    | parallel --no-run-if-empty -j 6 "
        echo '==> Group {}'

        if [ -e ${BASE_DIR}/{}/anchor/pe.anchor.fa ]; then
            exit;
        fi

        rm -fr ${BASE_DIR}/{}/anchor
        bash ~/Scripts/cpan/App-Anchr/share/anchor.sh ${BASE_DIR}/{} 8 false
    "

```

## Scer: results

* Stats of super-reads

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

REAL_G=12157105

bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 1 header \
    > ${BASE_DIR}/stat1.md

perl -e '
    for my $n (
        qw{
        original
        Q20L120 Q20L130 Q20L140 Q20L150
        Q25L120 Q25L130 Q25L140 Q25L150
        Q30L120 Q30L130 Q30L140 Q30L150
        }
        )
    {
        for my $i ( 1 .. 8 ) {
            printf qq{%s_%d\n}, $n, ( 1000000 * $i );
        }
    }
    ' \
    | parallel -k --no-run-if-empty -j 8 "
        if [ ! -d ${BASE_DIR}/{} ]; then
            exit;
        fi

        bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 1 ${BASE_DIR}/{} ${REAL_G}
    " >> ${BASE_DIR}/stat1.md

cat stat1.md
```

* Stats of anchors

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 2 header \
    > ${BASE_DIR}/stat2.md

perl -e '
    for my $n (
        qw{
        original
        Q20L120 Q20L130 Q20L140 Q20L150
        Q25L120 Q25L130 Q25L140 Q25L150
        Q30L120 Q30L130 Q30L140 Q30L150
        }
        )
    {
        for my $i ( 1 .. 8 ) {
            printf qq{%s_%d\n}, $n, ( 1000000 * $i );
        }
    }
    ' \
    | parallel -k --no-run-if-empty -j 16 "
        if [ ! -e ${BASE_DIR}/{}/anchor/pe.anchor.fa ]; then
            exit;
        fi

        bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 2 ${BASE_DIR}/{}
    " >> ${BASE_DIR}/stat2.md

cat stat2.md
```

| Name             |   SumFq | CovFq | AvgRead | Kmer |   SumFa | Discard% |  RealG |   EstG | Est/Real |  SumKU | SumSR |   RunTime |
|:-----------------|--------:|------:|--------:|-----:|--------:|---------:|-------:|-------:|---------:|-------:|------:|----------:|
| original_1000000 |    302M |  24.8 |     151 |  105 | 249.57M |  17.360% | 12.16M | 11.38M |     0.94 | 13.13M |     0 | 0:01'23'' |
| original_2000000 |    604M |  49.7 |     151 |  105 |  500.7M |  17.103% | 12.16M | 11.62M |     0.96 | 13.59M |     0 | 0:02'34'' |
| original_3000000 |    906M |  74.5 |     151 |  105 | 752.87M |  16.901% | 12.16M | 11.79M |     0.97 | 14.59M |     0 | 0:03'49'' |
| original_4000000 |   1.21G |  99.4 |     151 |  105 |   1.01G |  16.755% | 12.16M | 11.94M |     0.98 | 15.81M |     0 | 0:04'27'' |
| original_5000000 |   1.51G | 124.2 |     151 |  105 |   1.26G |  16.600% | 12.16M | 12.11M |     1.00 | 17.18M |     0 | 0:06'13'' |
| original_6000000 |   1.81G | 149.0 |     151 |  105 |   1.51G |  16.510% | 12.16M | 12.26M |     1.01 | 18.56M |     0 |           |
| original_7000000 |   2.11G | 173.9 |     151 |  105 |   1.77G |  16.431% | 12.16M | 12.42M |     1.02 | 20.07M |     0 |           |
| original_8000000 |   2.42G | 198.7 |     151 |  105 |   2.02G |  16.352% | 12.16M | 12.57M |     1.03 | 21.52M |     0 |           |
| Q20L120_1000000  | 300.68M |  24.7 |     150 |  105 | 273.06M |   9.183% | 12.16M | 11.35M |     0.93 | 12.78M |     0 | 0:01'41'' |
| Q20L120_2000000  | 601.35M |  49.5 |     150 |  105 | 547.02M |   9.035% | 12.16M | 11.52M |     0.95 | 12.73M |     0 | 0:02'39'' |
| Q20L120_3000000  | 902.03M |  74.2 |     150 |  105 | 820.63M |   9.024% | 12.16M | 11.58M |     0.95 |  12.9M |     0 | 0:03'52'' |
| Q20L120_4000000  |    1.2G |  98.9 |     150 |  105 |    1.1G |   8.937% | 12.16M | 11.65M |     0.96 | 13.38M |     0 | 0:05'02'' |
| Q20L120_5000000  |    1.5G | 123.7 |     150 |  105 |   1.37G |   8.848% | 12.16M | 11.72M |     0.96 | 13.99M |     0 | 0:05'53'' |
| Q20L120_6000000  |    1.8G | 148.4 |     150 |  105 |   1.65G |   8.778% | 12.16M |  11.8M |     0.97 | 14.63M |     0 |           |
| Q20L120_7000000  |    2.1G | 173.1 |     150 |  105 |   1.92G |   8.706% | 12.16M | 11.88M |     0.98 | 15.34M |     0 |           |
| Q20L120_8000000  |   2.41G | 197.9 |     150 |  105 |    2.2G |   8.654% | 12.16M | 11.96M |     0.98 | 16.06M |     0 |           |
| Q20L130_1000000  | 301.28M |  24.8 |     150 |  105 | 273.98M |   9.064% | 12.16M | 11.35M |     0.93 | 12.74M |     0 | 0:01'57'' |
| Q20L130_2000000  | 602.58M |  49.6 |     150 |  105 | 548.86M |   8.915% | 12.16M | 11.52M |     0.95 | 12.71M |     0 | 0:03'02'' |
| Q20L130_3000000  | 903.86M |  74.3 |     150 |  105 | 823.65M |   8.874% | 12.16M | 11.57M |     0.95 | 12.85M |     0 | 0:03'46'' |
| Q20L130_4000000  |   1.21G |  99.1 |     150 |  105 |    1.1G |   8.804% | 12.16M | 11.64M |     0.96 | 13.29M |     0 | 0:05'04'' |
| Q20L130_5000000  |   1.51G | 123.9 |     150 |  105 |   1.37G |   8.730% | 12.16M | 11.71M |     0.96 | 13.88M |     0 | 0:05'48'' |
| Q20L130_6000000  |   1.81G | 148.7 |     150 |  105 |   1.65G |   8.658% | 12.16M | 11.78M |     0.97 |  14.5M |     0 |           |
| Q20L130_7000000  |   2.11G | 173.5 |     150 |  105 |   1.93G |   8.593% | 12.16M | 11.86M |     0.98 | 15.19M |     0 |           |
| Q20L130_8000000  |   2.41G | 198.3 |     150 |  105 |    2.2G |   8.540% | 12.16M | 11.94M |     0.98 | 15.88M |     0 |           |
| Q20L140_1000000  | 301.78M |  24.8 |     150 |  105 | 275.22M |   8.803% | 12.16M | 11.34M |     0.93 | 12.69M |     0 | 0:01'47'' |
| Q20L140_2000000  | 603.57M |  49.6 |     150 |  105 | 551.02M |   8.707% | 12.16M | 11.51M |     0.95 | 12.65M |     0 | 0:02'56'' |
| Q20L140_3000000  | 905.36M |  74.5 |     150 |  105 | 826.62M |   8.697% | 12.16M | 11.56M |     0.95 | 12.77M |     0 | 0:03'48'' |
| Q20L140_4000000  |   1.21G |  99.3 |     150 |  105 |    1.1G |   8.627% | 12.16M | 11.62M |     0.96 | 13.19M |     0 | 0:05'03'' |
| Q20L140_5000000  |   1.51G | 124.1 |     150 |  105 |   1.38G |   8.558% | 12.16M | 11.68M |     0.96 |  13.7M |     0 | 0:06'12'' |
| Q20L140_6000000  |   1.81G | 148.9 |     150 |  105 |   1.66G |   8.478% | 12.16M | 11.75M |     0.97 | 14.29M |     0 |           |
| Q20L140_7000000  |   2.11G | 173.8 |     150 |  105 |   1.93G |   8.427% | 12.16M | 11.83M |     0.97 | 14.95M |     0 |           |
| Q20L140_8000000  |   2.41G | 198.6 |     150 |  105 |   2.21G |   8.378% | 12.16M |  11.9M |     0.98 |  15.6M |     0 |           |
| Q20L150_1000000  | 301.98M |  24.8 |     150 |  105 |  275.9M |   8.634% | 12.16M | 11.34M |     0.93 | 12.68M |     0 | 0:01'46'' |
| Q20L150_2000000  | 603.95M |  49.7 |     150 |  105 | 552.38M |   8.539% | 12.16M |  11.5M |     0.95 | 12.62M |     0 | 0:02'46'' |
| Q20L150_3000000  | 905.93M |  74.5 |     150 |  105 | 828.85M |   8.508% | 12.16M | 11.55M |     0.95 | 12.74M |     0 | 0:04'01'' |
| Q20L150_4000000  |   1.21G |  99.4 |     150 |  105 |   1.11G |   8.443% | 12.16M | 11.61M |     0.96 | 13.12M |     0 | 0:04'57'' |
| Q20L150_5000000  |   1.51G | 124.2 |     150 |  105 |   1.38G |   8.358% | 12.16M | 11.67M |     0.96 | 13.59M |     0 | 0:06'00'' |
| Q20L150_6000000  |   1.81G | 149.0 |     150 |  105 |   1.66G |   8.299% | 12.16M | 11.74M |     0.97 | 14.19M |     0 |           |
| Q20L150_7000000  |   2.11G | 173.9 |     150 |  105 |   1.94G |   8.241% | 12.16M | 11.81M |     0.97 |  14.8M |     0 |           |
| Q20L150_8000000  |   2.42G | 198.7 |     150 |  105 |   2.22G |   8.190% | 12.16M | 11.88M |     0.98 | 15.45M |     0 |           |
| Q25L120_1000000  | 300.59M |  24.7 |     150 |  105 | 280.59M |   6.651% | 12.16M | 11.34M |     0.93 |  12.7M |     0 | 0:01'40'' |
| Q25L120_2000000  | 601.18M |  49.5 |     150 |  105 | 561.44M |   6.610% | 12.16M | 11.49M |     0.95 | 12.44M |     0 | 0:02'48'' |
| Q25L120_3000000  | 901.77M |  74.2 |     150 |  105 | 842.61M |   6.560% | 12.16M | 11.54M |     0.95 | 12.58M |     0 | 0:03'55'' |
| Q25L120_4000000  |    1.2G |  98.9 |     150 |  105 |   1.12G |   6.526% | 12.16M | 11.58M |     0.95 | 12.83M |     0 | 0:05'01'' |
| Q25L120_5000000  |    1.5G | 123.6 |     150 |  105 |   1.41G |   6.466% | 12.16M | 11.61M |     0.96 | 13.15M |     0 | 0:06'01'' |
| Q25L120_6000000  |    1.8G | 148.4 |     150 |  105 |   1.69G |   6.423% | 12.16M | 11.66M |     0.96 | 13.56M |     0 |           |
| Q25L120_7000000  |    2.1G | 173.1 |     150 |  105 |   1.97G |   6.382% | 12.16M |  11.7M |     0.96 | 13.98M |     0 |           |
| Q25L120_8000000  |    2.4G | 197.8 |     150 |  105 |   2.25G |   6.348% | 12.16M | 11.75M |     0.97 | 14.47M |     0 |           |
| Q25L130_1000000  | 301.28M |  24.8 |     150 |  105 | 281.33M |   6.623% | 12.16M | 11.34M |     0.93 | 12.66M |     0 | 0:01'26'' |
| Q25L130_2000000  | 602.55M |  49.6 |     150 |  105 | 562.97M |   6.569% | 12.16M | 11.48M |     0.94 | 12.43M |     0 | 0:02'21'' |
| Q25L130_3000000  | 903.82M |  74.3 |     150 |  105 |  844.7M |   6.541% | 12.16M | 11.53M |     0.95 | 12.54M |     0 | 0:03'17'' |
| Q25L130_4000000  |   1.21G |  99.1 |     150 |  105 |   1.13G |   6.493% | 12.16M | 11.57M |     0.95 | 12.78M |     0 | 0:04'46'' |
| Q25L130_5000000  |   1.51G | 123.9 |     150 |  105 |   1.41G |   6.441% | 12.16M | 11.61M |     0.95 | 13.09M |     0 | 0:09'21'' |
| Q25L130_6000000  |   1.81G | 148.7 |     150 |  105 |   1.69G |   6.386% | 12.16M | 11.65M |     0.96 | 13.45M |     0 |           |
| Q25L130_7000000  |   2.11G | 173.5 |     150 |  105 |   1.97G |   6.356% | 12.16M | 11.69M |     0.96 |  13.9M |     0 |           |
| Q25L130_8000000  |   2.41G | 198.3 |     150 |  105 |   2.26G |   6.313% | 12.16M | 11.74M |     0.97 | 14.38M |     0 |           |
| Q25L140_1000000  | 301.81M |  24.8 |     150 |  105 | 282.02M |   6.559% | 12.16M | 11.33M |     0.93 | 12.61M |     0 | 0:02'09'' |
| Q25L140_2000000  | 603.62M |  49.7 |     150 |  105 | 564.34M |   6.507% | 12.16M | 11.48M |     0.94 | 12.39M |     0 | 0:04'11'' |
| Q25L140_3000000  | 905.42M |  74.5 |     150 |  105 | 846.76M |   6.479% | 12.16M | 11.53M |     0.95 | 12.51M |     0 | 0:05'44'' |
| Q25L140_4000000  |   1.21G |  99.3 |     150 |  105 |   1.13G |   6.429% | 12.16M | 11.56M |     0.95 | 12.73M |     0 | 0:07'22'' |
| Q25L140_5000000  |   1.51G | 124.1 |     150 |  105 |   1.41G |   6.374% | 12.16M |  11.6M |     0.95 | 13.02M |     0 | 0:09'22'' |
| Q25L140_6000000  |   1.81G | 149.0 |     150 |  105 |    1.7G |   6.332% | 12.16M | 11.64M |     0.96 | 13.38M |     0 |           |
| Q25L140_7000000  |   2.11G | 173.8 |     150 |  105 |   1.98G |   6.298% | 12.16M | 11.68M |     0.96 | 13.82M |     0 |           |
| Q25L150_1000000  | 301.99M |  24.8 |     150 |  105 | 282.36M |   6.499% | 12.16M | 11.32M |     0.93 |  12.6M |     0 | 0:02'13'' |
| Q25L150_2000000  | 603.98M |  49.7 |     150 |  105 | 564.85M |   6.479% | 12.16M | 11.48M |     0.94 | 12.39M |     0 | 0:04'10'' |
| Q25L150_3000000  | 905.97M |  74.5 |     150 |  105 | 847.77M |   6.423% | 12.16M | 11.52M |     0.95 |  12.5M |     0 | 0:05'54'' |
| Q25L150_4000000  |   1.21G |  99.4 |     150 |  105 |   1.13G |   6.412% | 12.16M | 11.56M |     0.95 |  12.7M |     0 | 0:07'39'' |
| Q25L150_5000000  |   1.51G | 124.2 |     150 |  105 |   1.41G |   6.347% | 12.16M | 11.59M |     0.95 | 12.99M |     0 | 0:09'22'' |
| Q25L150_6000000  |   1.81G | 149.0 |     150 |  105 |    1.7G |   6.305% | 12.16M | 11.63M |     0.96 | 13.37M |     0 |           |
| Q25L150_7000000  |   2.11G | 173.9 |     150 |  105 |   1.98G |   6.275% | 12.16M | 11.68M |     0.96 | 13.77M |     0 |           |
| Q30L120_1000000  | 300.37M |  24.7 |     150 |  105 | 283.43M |   5.640% | 12.16M | 11.33M |     0.93 | 12.63M |     0 | 0:02'30'' |
| Q30L120_2000000  | 600.73M |  49.4 |     150 |  105 |  567.1M |   5.598% | 12.16M | 11.47M |     0.94 | 12.37M |     0 | 0:03'34'' |
| Q30L120_3000000  | 901.09M |  74.1 |     150 |  105 | 850.76M |   5.586% | 12.16M | 11.52M |     0.95 | 12.45M |     0 | 0:05'58'' |
| Q30L120_4000000  |    1.2G |  98.8 |     150 |  105 |   1.13G |   5.546% | 12.16M | 11.55M |     0.95 | 12.63M |     0 | 0:07'12'' |
| Q30L120_5000000  |    1.5G | 123.5 |     150 |  105 |   1.42G |   5.516% | 12.16M | 11.58M |     0.95 | 12.87M |     0 | 0:09'51'' |
| Q30L120_6000000  |    1.8G | 148.2 |     150 |  105 |    1.7G |   5.475% | 12.16M | 11.61M |     0.96 | 13.15M |     0 |           |
| Q30L120_7000000  |    2.1G | 172.9 |     150 |  105 |   1.99G |   5.446% | 12.16M | 11.65M |     0.96 | 13.47M |     0 |           |
| Q30L130_1000000  | 301.16M |  24.8 |     150 |  105 | 284.13M |   5.655% | 12.16M | 11.33M |     0.93 |  12.6M |     0 | 0:02'14'' |
| Q30L130_2000000  | 602.31M |  49.5 |     150 |  105 | 568.56M |   5.603% | 12.16M | 11.47M |     0.94 | 12.36M |     0 | 0:04'50'' |
| Q30L130_3000000  | 903.47M |  74.3 |     150 |  105 | 853.15M |   5.570% | 12.16M | 11.51M |     0.95 | 12.44M |     0 | 0:05'38'' |
| Q30L130_4000000  |    1.2G |  99.1 |     150 |  105 |   1.14G |   5.542% | 12.16M | 11.54M |     0.95 |  12.6M |     0 | 0:07'31'' |
| Q30L130_5000000  |   1.51G | 123.9 |     150 |  105 |   1.42G |   5.504% | 12.16M | 11.58M |     0.95 | 12.81M |     0 | 0:09'13'' |
| Q30L130_6000000  |   1.81G | 148.6 |     150 |  105 |   1.71G |   5.466% | 12.16M | 11.61M |     0.95 | 13.11M |     0 |           |
| Q30L130_7000000  |   2.11G | 173.4 |     150 |  105 |   1.99G |   5.444% | 12.16M | 11.64M |     0.96 | 13.43M |     0 |           |
| Q30L140_1000000  | 301.78M |  24.8 |     150 |  105 | 284.85M |   5.611% | 12.16M | 11.31M |     0.93 | 12.56M |     0 | 0:02'32'' |
| Q30L140_2000000  | 603.56M |  49.6 |     150 |  105 | 569.81M |   5.592% | 12.16M | 11.46M |     0.94 | 12.33M |     0 | 0:03'58'' |
| Q30L140_3000000  | 905.35M |  74.5 |     150 |  105 | 855.07M |   5.554% | 12.16M |  11.5M |     0.95 | 12.39M |     0 | 0:04'51'' |
| Q30L140_4000000  |   1.21G |  99.3 |     150 |  105 |   1.14G |   5.531% | 12.16M | 11.54M |     0.95 | 12.56M |     0 | 0:07'47'' |
| Q30L140_5000000  |   1.51G | 124.1 |     150 |  105 |   1.43G |   5.492% | 12.16M | 11.57M |     0.95 | 12.78M |     0 | 0:08'50'' |
| Q30L140_6000000  |   1.81G | 148.9 |     150 |  105 |   1.71G |   5.469% | 12.16M |  11.6M |     0.95 | 13.03M |     0 |           |
| Q30L140_7000000  |   2.11G | 173.8 |     150 |  105 |      2G |   5.436% | 12.16M | 11.63M |     0.96 | 13.37M |     0 |           |
| Q30L150_1000000  | 301.99M |  24.8 |     150 |  105 | 285.18M |   5.566% | 12.16M | 11.31M |     0.93 | 12.55M |     0 | 0:02'22'' |
| Q30L150_2000000  | 603.99M |  49.7 |     150 |  105 | 570.39M |   5.563% | 12.16M | 11.46M |     0.94 | 12.32M |     0 | 0:03'38'' |
| Q30L150_3000000  | 905.98M |  74.5 |     150 |  105 | 855.95M |   5.522% | 12.16M |  11.5M |     0.95 | 12.39M |     0 | 0:06'32'' |
| Q30L150_4000000  |   1.21G |  99.4 |     150 |  105 |   1.14G |   5.493% | 12.16M | 11.53M |     0.95 | 12.54M |     0 | 0:06'58'' |
| Q30L150_5000000  |   1.51G | 124.2 |     150 |  105 |   1.43G |   5.458% | 12.16M | 11.56M |     0.95 | 12.75M |     0 | 0:09'29'' |
| Q30L150_6000000  |   1.81G | 149.0 |     150 |  105 |   1.71G |   5.426% | 12.16M |  11.6M |     0.95 | 13.02M |     0 |           |
| Q30L150_7000000  |    2.1G | 173.1 |     150 |  105 |   1.99G |   5.398% | 12.16M | 11.63M |     0.96 | 13.32M |     0 |           |

| Name             | N50SRclean |    Sum |     # | N50Anchor |    Sum |    # | N50Anchor2 |    Sum |  # | N50Others |   Sum |     # |   RunTime |
|:-----------------|-----------:|-------:|------:|----------:|-------:|-----:|-----------:|-------:|---:|----------:|------:|------:|----------:|
| original_1000000 |       1206 | 13.13M | 23247 |      1898 |  7.42M | 4038 |       1135 | 12.89K | 11 |       427 | 5.69M | 19198 | 0:01'30'' |
| original_2000000 |       3008 | 13.59M | 20451 |      3841 | 10.55M | 3336 |       1214 | 17.91K | 15 |       153 | 3.02M | 17100 | 0:02'28'' |
| original_3000000 |       3961 | 14.59M | 27172 |      5506 | 11.03M | 2646 |       1529 |  3.02K |  2 |       128 | 3.56M | 24524 | 0:03'07'' |
| original_4000000 |       4279 | 15.81M | 36461 |      6747 |  11.1M | 2310 |       1511 |  2.73K |  2 |       124 |  4.7M | 34149 | 0:04'24'' |
| original_5000000 |       3545 | 17.18M | 47415 |      6596 | 11.13M | 2363 |       1237 |  1.24K |  1 |       123 | 6.05M | 45051 | 0:04'36'' |
| original_6000000 |       2658 | 18.56M | 58381 |      6240 | 11.05M | 2491 |          0 |      0 |  0 |       123 | 7.51M | 55890 | 0:05'19'' |
| original_7000000 |       1761 | 20.07M | 70521 |      5294 | 10.96M | 2776 |          0 |      0 |  0 |       123 | 9.11M | 67745 | 0:06'25'' |
| original_8000000 |       1052 | 21.52M | 82117 |      4401 | 10.81M | 3159 |       1531 |   2.7K |  2 |       123 | 10.7M | 78956 | 0:06'49'' |
| Q20L120_1000000  |       1417 | 12.78M | 19079 |      2061 |  8.05M | 4126 |       1206 | 18.79K | 15 |       443 | 4.72M | 14938 | 0:01'32'' |
| Q20L120_2000000  |       3560 | 12.73M | 12669 |      4138 |  10.7M | 3173 |       1195 | 13.32K | 11 |       215 | 2.02M |  9485 | 0:01'52'' |
| Q20L120_3000000  |       5504 |  12.9M | 12527 |      6428 | 11.12M | 2365 |       1088 |  1.09K |  1 |       165 | 1.78M | 10161 | 0:02'20'' |
| Q20L120_4000000  |       6479 | 13.38M | 15887 |      7987 | 11.21M | 2035 |          0 |      0 |  0 |       144 | 2.17M | 13852 | 0:04'03'' |
| Q20L120_5000000  |       6744 | 13.99M | 20428 |      9107 | 11.26M | 1875 |          0 |      0 |  0 |       135 | 2.73M | 18553 | 0:04'34'' |
| Q20L120_6000000  |       6820 | 14.63M | 25305 |      9567 | 11.26M | 1832 |       1194 |  1.19K |  1 |       133 | 3.36M | 23472 | 0:05'02'' |
| Q20L120_7000000  |       6011 | 15.34M | 30821 |      9039 | 11.28M | 1882 |          0 |      0 |  0 |       131 | 4.06M | 28939 | 0:05'55'' |
| Q20L120_8000000  |       5475 | 16.06M | 36310 |      9158 | 11.26M | 1906 |          0 |      0 |  0 |       131 |  4.8M | 34404 | 0:06'47'' |
| Q20L130_1000000  |       1450 | 12.74M | 18623 |      2060 |  8.11M | 4144 |       1137 | 32.23K | 27 |       449 |  4.6M | 14452 | 0:01'05'' |
| Q20L130_2000000  |       3633 | 12.71M | 12408 |      4312 | 10.75M | 3175 |       1190 |  4.18K |  3 |       212 | 1.95M |  9230 | 0:02'04'' |
| Q20L130_3000000  |       5517 | 12.85M | 12117 |      6381 |  11.1M | 2386 |       1488 |  2.58K |  2 |       171 | 1.74M |  9729 | 0:02'24'' |
| Q20L130_4000000  |       6764 | 13.29M | 15082 |      8183 | 11.22M | 2000 |       1180 |  2.26K |  2 |       147 | 2.07M | 13080 | 0:03'27'' |
| Q20L130_5000000  |       6908 | 13.88M | 19474 |      9233 | 11.26M | 1860 |       1391 |  2.67K |  2 |       137 | 2.62M | 17612 | 0:04'24'' |
| Q20L130_6000000  |       6903 |  14.5M | 24202 |      9515 | 11.28M | 1822 |          0 |      0 |  0 |       134 | 3.22M | 22380 | 0:04'29'' |
| Q20L130_7000000  |       6335 | 15.19M | 29498 |      9314 | 11.28M | 1859 |          0 |      0 |  0 |       132 | 3.91M | 27639 | 0:05'46'' |
| Q20L130_8000000  |       5653 | 15.88M | 34859 |      9304 | 11.27M | 1892 |          0 |      0 |  0 |       132 | 4.62M | 32967 | 0:06'46'' |
| Q20L140_1000000  |       1482 | 12.69M | 18053 |      2128 |   8.2M | 4125 |       1237 | 23.36K | 19 |       454 | 4.47M | 13909 | 0:01'19'' |
| Q20L140_2000000  |       3700 | 12.65M | 11881 |      4261 | 10.78M | 3179 |       1236 |  2.41K |  2 |       217 | 1.87M |  8700 | 0:01'54'' |
| Q20L140_3000000  |       5680 | 12.77M | 11465 |      6617 | 11.13M | 2353 |          0 |      0 |  0 |       175 | 1.64M |  9112 | 0:03'01'' |
| Q20L140_4000000  |       6794 | 13.19M | 14264 |      8250 | 11.22M | 1990 |       1105 |  1.11K |  1 |       149 | 1.96M | 12273 | 0:03'27'' |
| Q20L140_5000000  |       7659 |  13.7M | 18073 |      9483 | 11.26M | 1805 |       1268 |  2.37K |  2 |       139 | 2.44M | 16266 | 0:03'53'' |
| Q20L140_6000000  |       7267 | 14.29M | 22574 |      9489 | 11.28M | 1794 |          0 |      0 |  0 |       135 | 3.01M | 20780 | 0:05'12'' |
| Q20L140_7000000  |       6928 | 14.95M | 27644 |      9622 | 11.28M | 1785 |          0 |      0 |  0 |       133 | 3.67M | 25859 | 0:05'41'' |
| Q20L140_8000000  |       5965 |  15.6M | 32559 |      9336 | 11.28M | 1846 |          0 |      0 |  0 |       132 | 4.32M | 30713 | 0:06'57'' |
| Q20L150_1000000  |       1493 | 12.68M | 17930 |      2118 |  8.28M | 4129 |       1153 | 15.17K | 13 |       441 | 4.38M | 13788 | 0:01'28'' |
| Q20L150_2000000  |       3726 | 12.62M | 11756 |      4312 | 10.75M | 3143 |       1140 |  3.46K |  3 |       224 | 1.87M |  8610 | 0:02'17'' |
| Q20L150_3000000  |       5583 | 12.74M | 11247 |      6422 | 11.12M | 2366 |       1308 |  3.76K |  3 |       180 | 1.62M |  8878 | 0:03'07'' |
| Q20L150_4000000  |       6814 | 13.12M | 13724 |      8190 | 11.23M | 1997 |       1692 |  2.92K |  2 |       151 | 1.89M | 11725 | 0:03'21'' |
| Q20L150_5000000  |       7655 | 13.59M | 17190 |      9446 | 11.26M | 1788 |       1105 |  1.11K |  1 |       141 | 2.33M | 15401 | 0:04'53'' |
| Q20L150_6000000  |       7344 | 14.19M | 21781 |      9905 | 11.27M | 1752 |       1553 |  1.55K |  1 |       136 | 2.92M | 20028 | 0:06'14'' |
| Q20L150_7000000  |       6848 |  14.8M | 26428 |      9691 | 11.29M | 1775 |          0 |      0 |  0 |       133 | 3.51M | 24653 | 0:06'31'' |
| Q20L150_8000000  |       6143 | 15.45M | 31390 |      9427 | 11.28M | 1833 |          0 |      0 |  0 |       133 | 4.16M | 29557 | 0:06'22'' |
| Q25L120_1000000  |       1463 |  12.7M | 18045 |      2079 |  8.21M | 4187 |       1213 | 14.28K | 12 |       457 | 4.47M | 13846 | 0:01'06'' |
| Q25L120_2000000  |       3852 | 12.44M | 10095 |      4379 |  10.8M | 3093 |       1194 |   6.7K |  5 |       248 | 1.63M |  6997 | 0:02'08'' |
| Q25L120_3000000  |       5753 | 12.58M |  9933 |      6481 | 11.12M | 2332 |       1228 |  4.66K |  4 |       198 | 1.46M |  7597 | 0:02'21'' |
| Q25L120_4000000  |       7161 | 12.83M | 11349 |      8447 | 11.22M | 1959 |       1274 |  1.27K |  1 |       162 | 1.61M |  9389 | 0:03'30'' |
| Q25L120_5000000  |       7920 | 13.15M | 13650 |      9444 | 11.25M | 1812 |          0 |      0 |  0 |       150 |  1.9M | 11838 | 0:04'36'' |
| Q25L120_6000000  |       8128 | 13.56M | 16734 |      9660 | 11.27M | 1741 |       1720 |  1.72K |  1 |       141 | 2.28M | 14992 | 0:05'26'' |
| Q25L120_7000000  |       7578 | 13.98M | 20022 |      9898 | 11.28M | 1768 |          0 |      0 |  0 |       138 |  2.7M | 18254 | 0:05'56'' |
| Q25L120_8000000  |       6899 | 14.47M | 23769 |      9600 | 11.28M | 1812 |       1720 |  1.72K |  1 |       136 | 3.19M | 21956 | 0:06'01'' |
| Q25L130_1000000  |       1514 | 12.66M | 17650 |      2095 |  8.32M | 4200 |       1165 | 29.29K | 24 |       450 | 4.31M | 13426 | 0:01'35'' |
| Q25L130_2000000  |       3858 | 12.43M | 10069 |      4408 | 10.76M | 3092 |       1185 |  3.68K |  3 |       264 | 1.67M |  6974 | 0:01'44'' |
| Q25L130_3000000  |       5716 | 12.54M |  9629 |      6453 | 11.14M | 2372 |       1132 |  2.22K |  2 |       203 | 1.41M |  7255 | 0:02'29'' |
| Q25L130_4000000  |       7347 | 12.78M | 10933 |      8292 | 11.24M | 1949 |       1236 |  1.24K |  1 |       164 | 1.54M |  8983 | 0:03'25'' |
| Q25L130_5000000  |       8240 | 13.09M | 13181 |      9685 | 11.27M | 1774 |       1109 |  1.11K |  1 |       149 | 1.82M | 11406 | 0:04'27'' |
| Q25L130_6000000  |       8263 | 13.45M | 15811 |     10281 | 11.29M | 1717 |          0 |      0 |  0 |       143 | 2.15M | 14094 | 0:05'27'' |
| Q25L130_7000000  |       7654 |  13.9M | 19329 |     10075 | 11.29M | 1753 |          0 |      0 |  0 |       139 | 2.61M | 17576 | 0:05'41'' |
| Q25L130_8000000  |       6923 | 14.38M | 23051 |      9622 | 11.29M | 1808 |          0 |      0 |  0 |       136 | 3.09M | 21243 | 0:06'06'' |
| Q25L140_1000000  |       1564 | 12.61M | 17108 |      2185 |  8.45M | 4156 |       1164 | 21.65K | 18 |       451 | 4.14M | 12934 | 0:01'31'' |
| Q25L140_2000000  |       3876 | 12.39M |  9760 |      4437 | 10.79M | 3045 |       1185 |   7.3K |  6 |       258 |  1.6M |  6709 | 0:01'44'' |
| Q25L140_3000000  |       5736 | 12.51M |  9360 |      6447 | 11.12M | 2326 |       1098 |  3.39K |  3 |       208 | 1.39M |  7031 | 0:02'31'' |
| Q25L140_4000000  |       7574 | 12.73M | 10596 |      8713 | 11.23M | 1910 |       1708 |  1.71K |  1 |       168 | 1.51M |  8685 | 0:03'38'' |
| Q25L140_5000000  |       8012 | 13.02M | 12598 |      9793 | 11.28M | 1805 |          0 |      0 |  0 |       151 | 1.74M | 10793 | 0:04'23'' |
| Q25L140_6000000  |       8502 | 13.38M | 15325 |     10179 | 11.29M | 1697 |          0 |      0 |  0 |       143 | 2.09M | 13628 | 0:04'41'' |
| Q25L140_7000000  |       7816 | 13.82M | 18711 |      9815 | 11.29M | 1742 |          0 |      0 |  0 |       139 | 2.53M | 16969 | 0:05'21'' |
| Q25L150_1000000  |       1571 |  12.6M | 17086 |      2141 |  8.42M | 4164 |       1148 | 21.71K | 18 |       455 | 4.16M | 12904 | 0:01'33'' |
| Q25L150_2000000  |       3956 | 12.39M |  9730 |      4536 | 10.81M | 3058 |       1123 |  5.71K |  5 |       257 | 1.58M |  6667 | 0:01'39'' |
| Q25L150_3000000  |       5919 |  12.5M |  9324 |      6807 | 11.15M | 2326 |       1199 |   1.2K |  1 |       200 | 1.35M |  6997 | 0:02'29'' |
| Q25L150_4000000  |       7447 |  12.7M | 10317 |      8631 | 11.24M | 1925 |          0 |      0 |  0 |       170 | 1.47M |  8392 | 0:05'05'' |
| Q25L150_5000000  |       8307 | 12.99M | 12375 |      9808 | 11.27M | 1782 |          0 |      0 |  0 |       151 | 1.71M | 10593 | 0:05'01'' |
| Q25L150_6000000  |       8595 | 13.37M | 15295 |     10291 | 11.29M | 1708 |       1109 |  1.11K |  1 |       143 | 2.08M | 13586 | 0:04'51'' |
| Q25L150_7000000  |       8024 | 13.77M | 18344 |     10134 | 11.29M | 1732 |          0 |      0 |  0 |       140 | 2.48M | 16612 | 0:05'48'' |
| Q30L120_1000000  |       1538 | 12.63M | 17375 |      2134 |  8.38M | 4183 |       1249 | 18.16K | 14 |       451 | 4.23M | 13178 | 0:01'19'' |
| Q30L120_2000000  |       3786 | 12.37M |  9721 |      4316 | 10.77M | 3120 |       1277 |  3.88K |  3 |       268 |  1.6M |  6598 | 0:01'56'' |
| Q30L120_3000000  |       5773 | 12.45M |  9005 |      6436 | 11.11M | 2325 |       1328 |  3.91K |  3 |       209 | 1.34M |  6677 | 0:02'38'' |
| Q30L120_4000000  |       7173 | 12.63M |  9828 |      7983 | 11.22M | 1990 |       1105 |  3.29K |  3 |       179 |  1.4M |  7835 | 0:03'43'' |
| Q30L120_5000000  |       7939 | 12.87M | 11477 |      9325 | 11.25M | 1832 |       1700 |  2.95K |  2 |       157 | 1.62M |  9643 | 0:04'41'' |
| Q30L120_6000000  |       8368 | 13.15M | 13545 |      9976 | 11.27M | 1744 |       1069 |  1.07K |  1 |       150 | 1.88M | 11800 | 0:04'49'' |
| Q30L120_7000000  |       8237 | 13.47M | 16017 |     10160 | 11.27M | 1742 |          0 |      0 |  0 |       144 |  2.2M | 14275 | 0:05'12'' |
| Q30L130_1000000  |       1550 |  12.6M | 17053 |      2140 |  8.36M | 4134 |       1170 | 14.64K | 12 |       472 | 4.23M | 12907 | 0:01'26'' |
| Q30L130_2000000  |       3879 | 12.36M |  9631 |      4394 | 10.76M | 3102 |       1132 |  1.13K |  1 |       278 |  1.6M |  6528 | 0:01'58'' |
| Q30L130_3000000  |       5658 | 12.44M |  8853 |      6421 |  11.1M | 2354 |          0 |      0 |  0 |       209 | 1.33M |  6499 | 0:03'14'' |
| Q30L130_4000000  |       7352 |  12.6M |  9652 |      8413 |  11.2M | 1979 |       1140 |  1.14K |  1 |       185 |  1.4M |  7672 | 0:03'56'' |
| Q30L130_5000000  |       8193 | 12.81M | 11009 |      9866 | 11.25M | 1790 |       1069 |  1.07K |  1 |       160 | 1.56M |  9218 | 0:04'32'' |
| Q30L130_6000000  |       8439 | 13.11M | 13288 |     10247 | 11.27M | 1730 |       1103 |   1.1K |  1 |       150 | 1.84M | 11557 | 0:04'55'' |
| Q30L130_7000000  |       8195 | 13.43M | 15669 |     10001 | 11.29M | 1747 |          0 |      0 |  0 |       144 | 2.14M | 13922 | 0:05'35'' |
| Q30L140_1000000  |       1580 | 12.56M | 16717 |      2185 |  8.46M | 4163 |       1167 | 11.92K | 10 |       458 | 4.09M | 12544 | 0:01'08'' |
| Q30L140_2000000  |       3889 | 12.33M |  9348 |      4420 | 10.79M | 3109 |       1336 |  7.59K |  6 |       278 | 1.53M |  6233 | 0:02'15'' |
| Q30L140_3000000  |       5864 | 12.39M |  8561 |      6515 |  11.1M | 2331 |       1103 |   1.1K |  1 |       209 | 1.29M |  6229 | 0:02'41'' |
| Q30L140_4000000  |       7366 | 12.56M |  9333 |      8473 | 11.21M | 1971 |          0 |      0 |  0 |       187 | 1.35M |  7362 | 0:03'35'' |
| Q30L140_5000000  |       8574 | 12.78M | 10807 |     10063 | 11.25M | 1786 |          0 |      0 |  0 |       161 | 1.53M |  9021 | 0:04'37'' |
| Q30L140_6000000  |       8504 | 13.03M | 12679 |      9890 | 11.28M | 1737 |          0 |      0 |  0 |       151 | 1.75M | 10942 | 0:04'43'' |
| Q30L140_7000000  |       8256 | 13.37M | 15229 |     10109 | 11.28M | 1731 |          0 |      0 |  0 |       145 | 2.08M | 13498 | 0:05'12'' |
| Q30L150_1000000  |       1591 | 12.55M | 16687 |      2182 |   8.5M | 4146 |       1359 | 17.19K | 13 |       455 | 4.03M | 12528 | 0:01'38'' |
| Q30L150_2000000  |       3913 | 12.32M |  9331 |      4389 | 10.78M | 3082 |       1308 |  7.63K |  6 |       276 | 1.53M |  6243 | 0:01'38'' |
| Q30L150_3000000  |       5837 | 12.39M |  8569 |      6468 | 11.12M | 2338 |       1110 |  1.11K |  1 |       209 | 1.27M |  6230 | 0:02'17'' |
| Q30L150_4000000  |       7367 | 12.54M |  9234 |      8458 | 11.22M | 1967 |       1089 |  1.09K |  1 |       188 | 1.33M |  7266 | 0:03'38'' |
| Q30L150_5000000  |       8404 | 12.75M | 10594 |      9685 | 11.26M | 1782 |          0 |      0 |  0 |       160 | 1.49M |  8812 | 0:04'20'' |
| Q30L150_6000000  |       8793 | 13.02M | 12626 |     10361 | 11.26M | 1695 |          0 |      0 |  0 |       151 | 1.76M | 10931 | 0:04'17'' |
| Q30L150_7000000  |       8261 | 13.32M | 14913 |     10109 | 11.28M | 1730 |          0 |      0 |  0 |       146 | 2.04M | 13183 | 0:05'12'' |

| Name             | N50SRclean |    Sum |     # | N50Anchor |    Sum |    # | N50Anchor2 |    Sum |  # | N50Others |   Sum |     # |   RunTime |
|:-----------------|-----------:|-------:|------:|----------:|-------:|-----:|-----------:|-------:|---:|----------:|------:|------:|----------:|
| original_4000000 |       4279 | 15.81M | 36461 |      6747 |  11.1M | 2310 |       1511 |  2.73K |  2 |       124 |  4.7M | 34149 | 0:04'24'' |
| Q20L120_6000000  |       6820 | 14.63M | 25305 |      9567 | 11.26M | 1832 |       1194 |  1.19K |  1 |       133 | 3.36M | 23472 | 0:05'02'' |
| Q20L130_6000000  |       6903 |  14.5M | 24202 |      9515 | 11.28M | 1822 |          0 |      0 |  0 |       134 | 3.22M | 22380 | 0:04'29'' |
| Q20L140_7000000  |       6928 | 14.95M | 27644 |      9622 | 11.28M | 1785 |          0 |      0 |  0 |       133 | 3.67M | 25859 | 0:05'41'' |
| Q20L150_6000000  |       7344 | 14.19M | 21781 |      9905 | 11.27M | 1752 |       1553 |  1.55K |  1 |       136 | 2.92M | 20028 | 0:06'14'' |
| Q25L120_7000000  |       7578 | 13.98M | 20022 |      9898 | 11.28M | 1768 |          0 |      0 |  0 |       138 |  2.7M | 18254 | 0:05'56'' |
| Q25L130_6000000  |       8263 | 13.45M | 15811 |     10281 | 11.29M | 1717 |          0 |      0 |  0 |       143 | 2.15M | 14094 | 0:05'27'' |
| Q25L140_6000000  |       8502 | 13.38M | 15325 |     10179 | 11.29M | 1697 |          0 |      0 |  0 |       143 | 2.09M | 13628 | 0:04'41'' |
| Q25L150_6000000  |       8595 | 13.37M | 15295 |     10291 | 11.29M | 1708 |       1109 |  1.11K |  1 |       143 | 2.08M | 13586 | 0:04'51'' |
| Q30L120_7000000  |       8237 | 13.47M | 16017 |     10160 | 11.27M | 1742 |          0 |      0 |  0 |       144 |  2.2M | 14275 | 0:05'12'' |
| Q30L130_7000000  |       8195 | 13.43M | 15669 |     10001 | 11.29M | 1747 |          0 |      0 |  0 |       144 | 2.14M | 13922 | 0:05'35'' |
| Q30L140_7000000  |       8256 | 13.37M | 15229 |     10109 | 11.28M | 1731 |          0 |      0 |  0 |       145 | 2.08M | 13498 | 0:05'12'' |
| Q30L150_7000000  |       8261 | 13.32M | 14913 |     10109 | 11.28M | 1730 |          0 |      0 |  0 |       146 | 2.04M | 13183 | 0:05'12'' |

## Scer: quality assessment

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

for part in anchor others;
do 
    bash ~/Scripts/cpan/App-Anchr/share/sort_on_ref.sh Q25L150_6000000/anchor/pe.${part}.fa 1_genome/genome.fa pe.${part}
    nucmer -l 200 1_genome/genome.fa pe.${part}.fa
    mummerplot -png out.delta -p pe.${part} --medium
done

cp ~/data/anchr/paralogs/model/Results/s288c/s288c.multi.fas 1_genome/paralogs.fas

# mummerplot files
rm *.[fr]plot
rm out.delta
rm *.gp

# merge anchors
mkdir merge
anchr contained \
    Q20L120_6000000/anchor/pe.anchor.fa \
    Q20L130_6000000/anchor/pe.anchor.fa \
    Q20L140_7000000/anchor/pe.anchor.fa \
    Q20L150_6000000/anchor/pe.anchor.fa \
    Q25L120_7000000/anchor/pe.anchor.fa \
    Q25L130_6000000/anchor/pe.anchor.fa \
    Q25L140_6000000/anchor/pe.anchor.fa \
    Q25L150_6000000/anchor/pe.anchor.fa \
    Q30L120_7000000/anchor/pe.anchor.fa \
    Q30L130_7000000/anchor/pe.anchor.fa \
    Q30L140_7000000/anchor/pe.anchor.fa \
    Q30L150_7000000/anchor/pe.anchor.fa \
    --len 1000 --idt 0.98 --proportion 0.99999 \
    -o stdout \
    | faops filter -a 1000 -l 0 stdin merge/anchor.contained.fasta
anchr orient merge/anchor.contained.fasta --len 1000 --idt 0.98 -o merge/anchor.orient.fasta
anchr merge merge/anchor.orient.fasta --len 1000 --idt 0.999 -o stdout \
    | faops filter -a 1000 -l 0 stdin merge/anchor.merge.fasta

faops n50 -S -C merge/anchor.merge.fasta

# quast
rm -fr 9_qa
quast --no-check \
    -R 1_genome/genome.fa \
    Q20L120_6000000/anchor/pe.anchor.fa \
    Q20L130_6000000/anchor/pe.anchor.fa \
    Q20L140_7000000/anchor/pe.anchor.fa \
    Q20L150_6000000/anchor/pe.anchor.fa \
    Q25L120_7000000/anchor/pe.anchor.fa \
    Q25L130_6000000/anchor/pe.anchor.fa \
    Q25L140_6000000/anchor/pe.anchor.fa \
    Q25L150_6000000/anchor/pe.anchor.fa \
    Q30L120_7000000/anchor/pe.anchor.fa \
    Q30L130_7000000/anchor/pe.anchor.fa \
    Q30L140_7000000/anchor/pe.anchor.fa \
    Q30L150_7000000/anchor/pe.anchor.fa \
    merge/anchor.merge.fasta \
    1_genome/paralogs.fas \
    --label "Q20L120,Q20L130,Q20L140,Q20L150,Q25L120,Q25L130,Q25L140,Q25L150,Q30L120,Q30L130,Q30L140,Q30L150,merge,paralogs" \
    -o 9_qa
```

## Scer: expand anchors

* anchorLong

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

head -n 230000 ${BASE_DIR}/3_pacbio/pacbio.fasta > ${BASE_DIR}/3_pacbio/pacbio.40x.fasta
faops n50 -S -C ${BASE_DIR}/3_pacbio/pacbio.40x.fasta

rm -fr covered
mkdir -p covered
anchr cover \
    -c 2 -m 60 \
    -b 20 --len 1000 --idt 0.8 \
    merge/anchor.merge.fasta \
    3_pacbio/pacbio.40x.fasta \
    -o covered/covered.fasta
faops n50 -S -C covered/covered.fasta

rm -fr anchorLong
anchr overlap2 \
    covered/covered.fasta \
    3_pacbio/pacbio.40x.fasta \
    -d anchorLong \
    -b 20 --len 1000 --idt 0.85

anchr overlap \
    ${BASE_DIR}/covered/covered.fasta \
    --serial --len 10 --idt 0.98 \
    -o stdout \
    | perl -nla -e '
        BEGIN {
            our %seen;
            our %count_of;
        }

        @F == 13 or next;
        $F[3] > 0.98 or next;

        my $pair = join( "-", sort { $a <=> $b } ( $F[0], $F[1], ) );
        next if $seen{$pair};
        $seen{$pair} = $_;

        $count_of{ $F[0] }++;
        $count_of{ $F[1] }++;

        END {
            for my $pair ( keys %seen ) {
                my ($f_id, $g_id) = split "-", $pair;
                next if $count_of{$f_id} > 2;
                next if $count_of{$g_id} > 2;
                print $seen{$pair};
            }
        }
    ' \
    | sort -k 1n,1n -k 2n,2n \
    > ${BASE_DIR}/anchorLong/anchor.ovlp.tsv

ANCHOR_COUNT=$(faops n50 -H -N 0 -C ${BASE_DIR}/anchorLong/anchor.fasta)
echo ${ANCHOR_COUNT}

rm -fr ${BASE_DIR}/anchorLong/group
anchr group \
    ${BASE_DIR}/anchorLong/anchorLong.db \
    ${BASE_DIR}/anchorLong/anchorLong.ovlp.tsv \
    --oa ${BASE_DIR}/anchorLong/anchor.ovlp.tsv \
    --range "1-${ANCHOR_COUNT}" --len 1000 --idt 0.85 --max 1 -c 4 --png

pushd ${BASE_DIR}/anchorLong
cat group/groups.txt \
    | parallel --no-run-if-empty -j 8 '
        echo {};
        anchr orient \
            --len 1000 --idt 0.85 \
            group/{}.anchor.fasta \
            group/{}.long.fasta \
            -r group/{}.restrict.tsv \
            -o group/{}.strand.fasta;

        anchr overlap --len 1000 --idt 0.85 \
            group/{}.strand.fasta \
            -o stdout \
            | anchr restrict \
                stdin group/{}.restrict.tsv \
                -o group/{}.ovlp.tsv;

        anchr overlap --len 10 --idt 0.98 \
            group/{}.strand.fasta \
            -o stdout \
            | perl -nla -e '\''
                @F == 13 or next;
                $F[3] > 0.98 or next;
                $F[9] == 0 or next;
                $F[5] > 0 and $F[6] == $F[7] or next;
                /anchor.+anchor/ or next;
                print;
            '\'' \
            > group/{}.anchor.ovlp.tsv
            
        anchr layout \
            group/{}.ovlp.tsv \
            group/{}.relation.tsv \
            group/{}.strand.fasta \
            --oa group/{}.anchor.ovlp.tsv \
            --png \
            -o group/{}.contig.fasta
    '
popd

# false strand
cat ${BASE_DIR}/anchorLong/group/*.ovlp.tsv \
    | perl -nla -e '/anchor.+long/ or next; print $F[0] if $F[8] == 1;' \
    | sort | uniq -c

faops n50 -S -C ${BASE_DIR}/anchorLong/group/*.contig.fasta

cat \
    ${BASE_DIR}/anchorLong/group/non_grouped.fasta\
    ${BASE_DIR}/anchorLong/group/*.contig.fasta \
    >  ${BASE_DIR}/anchorLong/contig.fasta
faops n50 -S -C ${BASE_DIR}/anchorLong/contig.fasta

```

* contigLong

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

rm -fr contigLong
anchr overlap2 \
    anchorLong/contig.fasta \
    3_pacbio/pacbio.40x.fasta \
    -d contigLong \
    -b 20 --len 2000 --idt 0.85 --all

CONTIG_COUNT=$(faops n50 -H -N 0 -C contigLong/anchor.fasta)
echo ${CONTIG_COUNT}
LONG_COUNT=$(faops n50 -H -N 0 -C contigLong/long.fasta)
echo ${LONG_COUNT}

# breaksLong
anchr break \
    contigLong/anchorLong.db \
    contigLong/anchorLong.ovlp.tsv \
    --range "1-${CONTIG_COUNT}" --len 2000 --idt 0.85 --power 1.1 \
    -o contigLong/breaksLong.fasta

# canu
rm -fr canu-breaks
canu \
    -correct \
    -p s288c -d canu-breaks \
    gnuplot=$(brew --prefix)/Cellar/$(brew list --versions gnuplot | sed 's/ /\//')/bin/gnuplot \
    genomeSize=12.2m \
    -pacbio-raw contigLong/breaksLong.fasta

canu \
    -trim \
    -p s288c -d canu-breaks \
    gnuplot=$(brew --prefix)/Cellar/$(brew list --versions gnuplot | sed 's/ /\//')/bin/gnuplot \
    genomeSize=12.2m \
    -pacbio-corrected canu-breaks/s288c.correctedReads.fasta.gz

faops n50 -S -C contigLong/breaksLong.fasta
faops n50 -S -C canu-breaks/s288c.trimmedReads.fasta.gz

```

* nonOverlappedLong

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

CONTIG_COUNT=$(faops n50 -H -N 0 -C contigLong/anchor.fasta)
echo ${CONTIG_COUNT}
LONG_COUNT=$(faops n50 -H -N 0 -C contigLong/long.fasta)
echo ${LONG_COUNT}

# nonOverlappedLong
cat contigLong/anchorLong.ovlp.tsv \
    | CONTIG_COUNT=${CONTIG_COUNT} perl -nla -e '
        BEGIN {
            our %seen;
        }

        @F == 13 or next;

        my $pair = join( "-", sort { $a <=> $b } ( $F[0], $F[1], ) );
        next if $seen{$pair};
        $seen{$pair} = $_;

        if ( $F[0] <= $ENV{CONTIG_COUNT} and $F[1] > $ENV{CONTIG_COUNT} ) {
            print $F[1];
        }
    ' \
    | sort -n | uniq \
    > contigLong/overlappedLong.serial.txt

grep -Fx -v \
    -f contigLong/overlappedLong.serial.txt \
    <(seq $((${CONTIG_COUNT} + 1)) 1 ${LONG_COUNT}) \
    > contigLong/nonOverlappedLong.serial.txt

DBshow -n contigLong/anchorLong.db \
    contigLong/nonOverlappedLong.serial.txt \
    | sed 's/^>//' \
    > contigLong/nonOverlappedLong.header.txt

faops some -l 0 \
    contigLong/long.fasta \
    contigLong/nonOverlappedLong.header.txt \
    contigLong/nonOverlappedLong.fasta

faops n50 -S -C contigLong/nonOverlappedLong.fasta

```

* contigTrim

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

rm -fr contigTrim
anchr overlap2 \
    anchorLong/contig.fasta \
    canu-breaks/s288c.trimmedReads.fasta.gz \
    -d contigTrim \
    -b 10 --len 1000 --idt 0.96

CONTIG_COUNT=$(faops n50 -H -N 0 -C contigTrim/anchor.fasta)
echo ${CONTIG_COUNT}

rm -fr ${BASE_DIR}/contigTrim/group
anchr group \
    contigTrim/anchorLong.db \
    contigTrim/anchorLong.ovlp.tsv \
    --range "1-${CONTIG_COUNT}" --len 1000 --idt 0.96 --max 5000 -c 8

pushd ${BASE_DIR}/contigTrim
cat group/groups.txt \
    | parallel --no-run-if-empty -j 8 '
        echo {};
        anchr orient \
            --len 1000 --idt 0.96 \
            group/{}.anchor.fasta \
            group/{}.long.fasta \
            -r group/{}.restrict.tsv \
            -o group/{}.strand.fasta;

        anchr overlap --len 1000 --idt 0.96 \
            group/{}.strand.fasta \
            -o stdout \
            | anchr restrict \
                stdin group/{}.restrict.tsv \
                -o group/{}.ovlp.tsv;

        anchr layout \
            group/{}.ovlp.tsv \
            group/{}.relation.tsv \
            group/{}.strand.fasta \
            -o group/{}.contig.fasta
    '
popd

faops n50 -S -C contigTrim/group/*.contig.fasta

cat \
    contigTrim/group/non_grouped.fasta \
    contigTrim/group/*.contig.fasta \
    >  contigTrim/contig.fasta
faops n50 -S -C contigTrim/contig.fasta

```

* contigFinal

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

rm -fr contigFinal
anchr overlap2 \
    contigTrim/contig.fasta \
    canu-breaks/s288c.trimmedReads.fasta.gz \
    -d contigFinal \
    -b 10 --len 1000 --idt 0.96

CONTIG_COUNT=$(faops n50 -H -N 0 -C contigFinal/anchor.fasta)
echo ${CONTIG_COUNT}
LONG_COUNT=$(faops n50 -H -N 0 -C contigFinal/long.fasta)
echo ${LONG_COUNT}

# nonContainedLong
cat contigFinal/anchorLong.ovlp.tsv \
    | CONTIG_COUNT=${CONTIG_COUNT} perl -nla -e '
        BEGIN {
            our %seen;
        }

        @F == 13 or next;

        my $pair = join( "-", sort { $a <=> $b } ( $F[0], $F[1], ) );
        next if $seen{$pair};
        $seen{$pair} = $_;

        if ( $F[0] <= $ENV{CONTIG_COUNT} and $F[1] > $ENV{CONTIG_COUNT} ) {
            if ( $F[12] eq "overlap" ) {
                print $F[1];
            }
        }
    ' \
    | sort -n | uniq \
    > contigFinal/nonContainedLong.serial.txt

DBshow -n contigFinal/anchorLong.db \
    contigFinal/nonContainedLong.serial.txt \
    | sed 's/^>//' \
    > contigFinal/nonContainedLong.header.txt

faops some -l 0 \
    contigFinal/long.fasta \
    contigFinal/nonContainedLong.header.txt \
    contigFinal/nonContainedLong.fasta

faops n50 -S -C contigFinal/nonContainedLong.fasta

# canu
rm -fr canu-non-contained
canu \
    -correct \
    -p s288c -d canu-non-contained \
    gnuplot=$(brew --prefix)/Cellar/$(brew list --versions gnuplot | sed 's/ /\//')/bin/gnuplot \
    genomeSize=12.2m \
    -pacbio-raw \
    contigFinal/nonContainedLong.fasta \
    contigLong/nonOverlappedLong.fasta

canu \
    -trim \
    -p s288c -d canu-non-contained \
    gnuplot=$(brew --prefix)/Cellar/$(brew list --versions gnuplot | sed 's/ /\//')/bin/gnuplot \
    genomeSize=12.2m \
    -pacbio-corrected canu-non-contained/s288c.correctedReads.fasta.gz

faops n50 -S -C canu-non-contained/s288c.trimmedReads.fasta.gz

rm -fr canu-anchor
canu \
    -assemble \
    -p s288c -d canu-anchor \
    gnuplot=$(brew --prefix)/Cellar/$(brew list --versions gnuplot | sed 's/ /\//')/bin/gnuplot \
    genomeSize=12.2m \
    -pacbio-corrected \
    contigFinal/anchor.fasta \
    contigFinal/anchor.fasta \
    canu-non-contained/s288c.trimmedReads.fasta.gz

rm -fr canu-anchor2
canu \
    -assemble \
    -p s288c -d canu-anchor2 \
    gnuplot=$(brew --prefix)/Cellar/$(brew list --versions gnuplot | sed 's/ /\//')/bin/gnuplot \
    genomeSize=12.2m \
    -pacbio-corrected \
    contigFinal/anchor.fasta \
    contigFinal/anchor.fasta \
    contigFinal/nonContainedLong.fasta
```

```bash
BASE_DIR=$HOME/data/anchr/s288c
cd ${BASE_DIR}

canu \
    -p s288c -d canu-raw-40x \
    gnuplot=$(brew --prefix)/Cellar/$(brew list --versions gnuplot | sed 's/ /\//')/bin/gnuplot \
    genomeSize=12.2m \
    -pacbio-raw 3_pacbio/pacbio.40x.fasta

canu \
    -p s288c -d canu-raw-all \
    gnuplot=$(brew --prefix)/Cellar/$(brew list --versions gnuplot | sed 's/ /\//')/bin/gnuplot \
    genomeSize=12.2m \
    -pacbio-raw 3_pacbio/pacbio.fasta

rm -fr 9_qa_contig
quast --no-check \
    -R 1_genome/genome.fa \
    merge/anchor.merge.fasta \
    anchorLong/contig.fasta \
    contigTrim/contig.fasta \
    canu-anchor/s288c.contigs.fasta \
    canu-anchor2/s288c.contigs.fasta \
    canu-raw-40x/s288c.unitigs.fasta \
    canu-raw-all/s288c.unitigs.fasta \
    1_genome/paralogs.fas \
    --label "merge,contig,contigTrim,canu-anchor,canu-anchor2,canu-40x,canu-all,paralogs" \
    -o 9_qa_contig

```

# *Drosophila melanogaster* iso-1

* Genome: [Ensembl 82](http://sep2015.archive.ensembl.org/Drosophila_melanogaster/Info/Index)
* Proportion of paralogs (> 1000 bp): 0.0661

## Dmel: download

* Reference genome

```bash
mkdir -p ~/data/anchr/iso_1/1_genome
cd ~/data/anchr/iso_1/1_genome

wget -N ftp://ftp.ensembl.org/pub/release-82/fasta/drosophila_melanogaster/dna/Drosophila_melanogaster.BDGP6.dna_sm.toplevel.fa.gz
faops order Drosophila_melanogaster.BDGP6.dna_sm.toplevel.fa.gz \
    <(for chr in {2L,2R,3L,3R,4,X,Y,dmel_mitochondrion_genome}; do echo $chr; done) \
    genome.fa
```

* Illumina

    SRR306628 labels ycnbwsp instead of iso-1.

```bash
# Downloading from ena with aria2
mkdir -p ~/data/anchr/iso_1/2_illumina
cd ~/data/anchr/iso_1/2_illumina
aria2c -x 9 -s 3 -c ftp://ftp.sra.ebi.ac.uk/vol1/srr/SRR306/SRR306628
fastq-dump --split-files ./SRR306628  
find . -name "*.fastq" | parallel -j 2 pigz -p 8

ln -s SRR306628_1.fastq.gz R1.fq.gz
ln -s SRR306628_2.fastq.gz R2.fq.gz
```

* PacBio

```bash
mkdir -p ~/data/anchr/iso_1/3_pacbio

```

## Dmel: trim

* Q20L120

```bash
mkdir -p ~/data/anchr/iso_1/2_illumina/Q20L120
pushd ~/data/anchr/iso_1/2_illumina/Q20L120

anchr trim \
    -q 20 -l 120 \
    ../R1.fq.gz ../R2.fq.gz \
    -o stdout \
    | bash
    
popd
```

* Stats

```bash
cd ~/data/anchr/iso_1

printf "| %s | %s | %s | %s |\n" \
    "Name" "N50" "Sum" "#" \
    > stat.md
printf "|:--|--:|--:|--:|\n" >> stat.md

printf "| %s | %s | %s | %s |\n" \
    $(echo "Genome";   faops n50 -H -S -C 1_genome/genome.fa;) >> stat.md
printf "| %s | %s | %s | %s |\n" \
    $(echo "Illumina"; faops n50 -H -S -C 2_illumina/R1.fq.gz 2_illumina/R2.fq.gz;) >> stat.md
printf "| %s | %s | %s | %s |\n" \
    $(echo "PacBio";   faops n50 -H -S -C 3_pacbio/pacbio.fasta;) >> stat.md

printf "| %s | %s | %s | %s |\n" \
    $(echo "Q20L120";  faops n50 -H -S -C 2_illumina/Q20L120/R1.fq.gz 2_illumina/Q20L120/R1.fq.gz;) >> stat.md

cat stat.md
```

| Name     |      N50 |         Sum |        # |
|:---------|---------:|------------:|---------:|
| Genome   | 25286936 |   137567477 |        8 |
| Illumina |      146 | 12852672000 | 88032000 |
| PacBio   |          |             |          |
| Q20L120  |      146 |  7346135978 | 51441736 |

## Dmel: down sampling

```bash
BASE_DIR=$HOME/data/anchr/iso_1
cd ${BASE_DIR}

# works on bash 3
ARRAY=( "2_illumina/Q20L120:Q20L120:25000000")

for group in "${ARRAY[@]}" ; do
    
    GROUP_DIR=$(group=${group} perl -e '@p = split q{:}, $ENV{group}; print $p[0];')
    GROUP_ID=$( group=${group} perl -e '@p = split q{:}, $ENV{group}; print $p[1];')
    GROUP_MAX=$(group=${group} perl -e '@p = split q{:}, $ENV{group}; print $p[2];')
    printf "==> %s \t %s \t %s\n" "$GROUP_DIR" "$GROUP_ID" "$GROUP_MAX"

    for count in $(perl -e 'print 5000000 * $_, q{ } for 1 .. 5');
    do
        if [[ "$count" -gt "$GROUP_MAX" ]]; then
            continue     
        fi
        
        echo "==> Reads ${GROUP_ID}_${count}"
        DIR_COUNT="${BASE_DIR}/${GROUP_ID}_${count}"
        mkdir -p ${DIR_COUNT}
        
        if [ -e ${DIR_COUNT}/R1.fq.gz ]; then
            continue     
        fi
        
        seqtk sample -s${count} \
            ${BASE_DIR}/${GROUP_DIR}/R1.fq.gz ${count} \
            | pigz > ${DIR_COUNT}/R1.fq.gz
        seqtk sample -s${count} \
            ${BASE_DIR}/${GROUP_DIR}/R2.fq.gz ${count} \
            | pigz > ${DIR_COUNT}/R2.fq.gz
    done
done
```

## Dmel: generate super-reads

```bash
BASE_DIR=$HOME/data/anchr/iso_1
cd ${BASE_DIR}

for d in $(perl -e 'for $n (qw{Q20L120}) { for $i (1 .. 5) { printf qq{%s_%d }, $n, (5000000 * $i); } }');
do
    echo
    echo "==> Reads ${d}"
    DIR_COUNT="${BASE_DIR}/${d}/"

    if [ ! -d ${DIR_COUNT} ]; then
        continue
    fi
    
    if [ -e ${DIR_COUNT}/pe.cor.fa ]; then
        echo "    pe.cor.fa already presents"
        continue
    fi
    
    pushd ${DIR_COUNT} > /dev/null
    anchr superreads \
        R1.fq.gz \
        R2.fq.gz \
        --nosr \
        -s 335 -d 33 -p 16
    bash superreads.sh
    popd > /dev/null
done
```

Clear intermediate files.

```bash
cd $HOME/data/anchr/iso_1/

find . -type f -name "quorum_mer_db.jf" | xargs rm
find . -type f -name "k_u_hash_0" | xargs rm
find . -type f -name "readPositionsInSuperReads" | xargs rm
find . -type f -name "*.tmp" | xargs rm
#find . -type f -name "pe.renamed.fastq" | xargs rm
```

## Dmel: create anchors

```bash
BASE_DIR=$HOME/data/anchr/iso_1
cd ${BASE_DIR}

for d in $(perl -e 'for $n (qw{Q20L120}) { for $i (1 .. 5) { printf qq{%s_%d }, $n, (5000000 * $i); } }');
do
    echo
    echo "==> Reads ${d}"
    DIR_COUNT="${BASE_DIR}/${d}/"
    
    if [ -e ${DIR_COUNT}/anchor/pe.anchor.fa ]; then
        continue
    fi
    
    rm -fr ${DIR_COUNT}/anchor
    bash ~/Scripts/cpan/App-Anchr/share/anchor.sh ${DIR_COUNT} 16 120 false
done
```

## Dmel: results

* Stats of super-reads

```bash
BASE_DIR=$HOME/data/anchr/iso_1
cd ${BASE_DIR}

REAL_G=137567477

bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 1 header \
    > ${BASE_DIR}/stat1.md

for d in $(perl -e 'for $n (qw{Q20L120}) { for $i (1 .. 5) { printf qq{%s_%d }, $n, (5000000 * $i); } }');
do
    DIR_COUNT="${BASE_DIR}/${d}/"
    
    if [ ! -d ${DIR_COUNT} ]; then
        continue     
    fi
    
    bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 1 ${DIR_COUNT} ${REAL_G} \
        >> ${BASE_DIR}/stat1.md
done

cat stat1.md
```

* Stats of anchors

```bash
BASE_DIR=$HOME/data/anchr/iso_1
cd ${BASE_DIR}

bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 2 header \
    > ${BASE_DIR}/stat2.md

for d in $(perl -e 'for $n (qw{Q20L120}) { for $i (1 .. 5) { printf qq{%s_%d }, $n, (5000000 * $i); } }');
do
    DIR_COUNT="${BASE_DIR}/${d}/"
    
    if [ ! -e ${DIR_COUNT}/anchor/pe.anchor.fa ]; then
        continue     
    fi
    
    bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 2 ${DIR_COUNT} \
        >> ${BASE_DIR}/stat2.md

done

cat stat2.md
```

| Name             | SumFq | CovFq | AvgRead | Kmer | SumFa | Discard% |   RealG |    EstG | Est/Real |   SumKU | SumSR |   RunTime |
|:-----------------|------:|------:|--------:|-----:|------:|---------:|--------:|--------:|---------:|--------:|------:|----------:|
| Q20L120_5000000  | 1.42G |  10.3 |     141 |   95 | 1.14G |  19.452% | 137.57M |  94.68M |     0.69 | 119.61M |     0 | 0:08'22'' |
| Q20L120_10000000 | 2.83G |  20.6 |     140 |   93 | 2.31G |  18.478% | 137.57M | 111.82M |     0.81 | 137.92M |     0 | 0:15'14'' |
| Q20L120_15000000 | 4.25G |  30.9 |     139 |   93 | 3.48G |  18.123% | 137.57M | 119.21M |     0.87 | 148.11M |     0 | 0:21'59'' |
| Q20L120_20000000 | 5.67G |  41.2 |     138 |   91 | 4.66G |  17.892% | 137.57M | 123.85M |     0.90 | 155.73M |     0 | 0:29'07'' |
| Q20L120_25000000 | 7.09G |  51.5 |     137 |   91 | 5.83G |  17.713% | 137.57M | 127.21M |     0.92 | 163.69M |     0 | 0:46'16'' |

| Name             | N50SRclean |     Sum |     # | N50Anchor |    Sum |     # | N50Anchor2 |     Sum |   # | N50Others |    Sum |     # |   RunTime |
|:-----------------|-----------:|--------:|------:|----------:|-------:|------:|-----------:|--------:|----:|----------:|-------:|------:|----------:|
| Q20L120_5000000  |        903 |  41.22M | 45797 |      1577 |  16.8M | 10364 |       1161 | 230.93K | 195 |       693 | 24.19M | 35238 | 0:06'18'' |
| Q20L120_10000000 |       1386 |  78.06M | 64045 |      2025 | 50.71M | 25913 |       1178 | 360.82K | 298 |       734 | 26.99M | 37834 | 0:10'51'' |
| Q20L120_15000000 |       1783 |  92.63M | 63874 |      2375 | 69.11M | 31492 |       1172 | 302.73K | 247 |       749 | 23.21M | 32135 | 0:14'08'' |
| Q20L120_20000000 |       2185 | 101.08M | 60715 |      2710 | 81.37M | 33713 |       1186 | 240.58K | 196 |       755 | 19.46M | 26806 | 0:18'30'' |
| Q20L120_25000000 |       2379 | 105.54M | 59727 |      2887 |  87.2M | 34576 |       1198 | 178.44K | 145 |       754 | 18.16M | 25006 | 0:26'37'' |

## Dmel: quality assessment

```bash
BASE_DIR=$HOME/data/anchr/iso_1
cd ${BASE_DIR}

for part in anchor anchor2 others;
do 
    bash ~/Scripts/cpan/App-Anchr/share/sort_on_ref.sh Q20L120_25000000/anchor/pe.${part}.fa 1_genome/genome.fa pe.${part}
    nucmer -l 200 1_genome/genome.fa pe.${part}.fa
    mummerplot -png out.delta -p pe.${part} --medium
done

# mummerplot files
rm *.[fr]plot
rm out.delta
rm *.gp

cp ~/data/anchr/paralogs/model/Results/iso_1/iso_1.multi.fas 1_genome/paralogs.fas

# quast
rm -fr 9_qa
quast --no-check \
    -R 1_genome/genome.fa \
    Q20L120_25000000/anchor/pe.anchor.fa \
    1_genome/paralogs.fas \
    --label "25000000,paralogs" \
    -o 9_qa
```

# *Caenorhabditis elegans* N2

* Genome: [Ensembl 82](http://sep2015.archive.ensembl.org/Caenorhabditis_elegans/Info/Index)
* Proportion of paralogs (> 1000 bp): 0.0472

## Cele: download

* Reference genome

```bash
mkdir -p ~/data/anchr/n2/1_genome
cd ~/data/anchr/n2/1_genome

wget -N ftp://ftp.ensembl.org/pub/release-82/fasta/caenorhabditis_elegans/dna/Caenorhabditis_elegans.WBcel235.dna_sm.toplevel.fa.gz
faops order Caenorhabditis_elegans.WBcel235.dna_sm.toplevel.fa.gz \
    <(for chr in {I,II,III,IV,V,X,MtDNA}; do echo $chr; done) \
    genome.fa
```

* Illumina

    * Other SRA
        * SRX770040 - [insert size](https://www.ncbi.nlm.nih.gov/sra/SRX770040[accn]) is 500-600 bp
        * ERR1039478 - adaptor contamination "ACTTCCAGGGATTTATAAGCCGATGACGTCATAACATCCCTGACCCTTTA"
        * DRR008443

```bash
# Downloading from ena with aria2
mkdir -p ~/data/anchr/n2/2_illumina
cd ~/data/anchr/n2/2_illumina
aria2c -x 9 -s 3 -c ftp://ftp.sra.ebi.ac.uk/vol1/srr/SRR065/SRR065390
fastq-dump --split-files ./SRR065390  
find . -name "*.fastq" | parallel -j 2 pigz -p 8

ln -s SRR065390_1.fastq.gz R1.fq.gz
ln -s SRR065390_2.fastq.gz R2.fq.gz
```

* PacBio

```bash
mkdir -p ~/data/anchr/n2/3_pacbio

```

## Cele: trim

* Q20L80

```bash
mkdir -p ~/data/anchr/n2/2_illumina/Q20L80
pushd ~/data/anchr/n2/2_illumina/Q20L80

anchr trim \
    -q 20 -l 80 \
    ../R1.fq.gz ../R2.fq.gz \
    -o stdout \
    | bash

popd
```

* Stats

```bash
cd ~/data/anchr/n2

printf "| %s | %s | %s | %s |\n" \
    "Name" "N50" "Sum" "#" \
    > stat.md
printf "|:--|--:|--:|--:|\n" >> stat.md

printf "| %s | %s | %s | %s |\n" \
    $(echo "Genome";   faops n50 -H -S -C 1_genome/genome.fa;) >> stat.md
printf "| %s | %s | %s | %s |\n" \
    $(echo "Illumina"; faops n50 -H -S -C 2_illumina/R1.fq.gz 2_illumina/R2.fq.gz;) >> stat.md
printf "| %s | %s | %s | %s |\n" \
    $(echo "PacBio";   faops n50 -H -S -C 3_pacbio/pacbio.fasta;) >> stat.md

printf "| %s | %s | %s | %s |\n" \
    $(echo "Q20L80";  faops n50 -H -S -C 2_illumina/Q20L80/R1.fq.gz 2_illumina/Q20L80/R1.fq.gz;) >> stat.md

cat stat.md
```

| Name     |      N50 |        Sum |        # |
|:---------|---------:|-----------:|---------:|
| Genome   | 17493829 |  100286401 |        7 |
| Illumina |      100 | 6761709200 | 67617092 |
| PacBio   |          |            |          |
| Q20L80   |      100 | 4771828790 | 48565296 |

## Cele: down sampling

```bash
BASE_DIR=$HOME/data/anchr/n2
cd ${BASE_DIR}

# works on bash 3
ARRAY=( "2_illumina/Q20L80:Q20L80:25000000")

for group in "${ARRAY[@]}" ; do
    
    GROUP_DIR=$(group=${group} perl -e '@p = split q{:}, $ENV{group}; print $p[0];')
    GROUP_ID=$( group=${group} perl -e '@p = split q{:}, $ENV{group}; print $p[1];')
    GROUP_MAX=$(group=${group} perl -e '@p = split q{:}, $ENV{group}; print $p[2];')
    printf "==> %s \t %s \t %s\n" "$GROUP_DIR" "$GROUP_ID" "$GROUP_MAX"

    for count in $(perl -e 'print 5000000 * $_, q{ } for 1 .. 5');
    do
        if [[ "$count" -gt "$GROUP_MAX" ]]; then
            continue     
        fi
        
        echo "==> Reads ${GROUP_ID}_${count}"
        DIR_COUNT="${BASE_DIR}/${GROUP_ID}_${count}"
        mkdir -p ${DIR_COUNT}
        
        if [ -e ${DIR_COUNT}/R1.fq.gz ]; then
            continue     
        fi
        
        seqtk sample -s${count} \
            ${BASE_DIR}/${GROUP_DIR}/R1.fq.gz ${count} \
            | pigz > ${DIR_COUNT}/R1.fq.gz
        seqtk sample -s${count} \
            ${BASE_DIR}/${GROUP_DIR}/R2.fq.gz ${count} \
            | pigz > ${DIR_COUNT}/R2.fq.gz
    done
done
```

## Cele: generate super-reads

```bash
BASE_DIR=$HOME/data/anchr/n2
cd ${BASE_DIR}

for d in $(perl -e 'for $n (qw{Q20L80}) { for $i (1 .. 5) { printf qq{%s_%d }, $n, (5000000 * $i); } }');
do
    echo
    echo "==> Reads ${d}"
    DIR_COUNT="${BASE_DIR}/${d}/"

    if [ ! -d ${DIR_COUNT} ]; then
        continue
    fi
    
    if [ -e ${DIR_COUNT}/pe.cor.fa ]; then
        echo "    pe.cor.fa already presents"
        continue
    fi
    
    pushd ${DIR_COUNT} > /dev/null
    anchr superreads \
        R1.fq.gz \
        R2.fq.gz \
        --nosr \
        -s 200 -d 20 -p 16
    bash superreads.sh
    popd > /dev/null
done
```

Clear intermediate files.

```bash
cd $HOME/data/anchr/n2/

find . -type f -name "quorum_mer_db.jf" | xargs rm
find . -type f -name "k_u_hash_0" | xargs rm
find . -type f -name "readPositionsInSuperReads" | xargs rm
find . -type f -name "*.tmp" | xargs rm
#find . -type f -name "pe.renamed.fastq" | xargs rm
```

## Cele: create anchors

```bash
BASE_DIR=$HOME/data/anchr/n2
cd ${BASE_DIR}

for d in $(perl -e 'for $n (qw{Q20L80}) { for $i (1 .. 5) { printf qq{%s_%d }, $n, (5000000 * $i); } }');
do
    echo
    echo "==> Reads ${d}"
    DIR_COUNT="${BASE_DIR}/${d}/"
    
    if [ -e ${DIR_COUNT}/anchor/pe.anchor.fa ]; then
        continue
    fi
    
    rm -fr ${DIR_COUNT}/anchor
    bash ~/Scripts/cpan/App-Anchr/share/anchor.sh ${DIR_COUNT} 16 80 false
done
```

## Cele: results

* Stats of super-reads

```bash
BASE_DIR=$HOME/data/anchr/n2
cd ${BASE_DIR}

REAL_G=100286401

bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 1 header \
    > ${BASE_DIR}/stat1.md

for d in $(perl -e 'for $n (qw{Q20L80}) { for $i (1 .. 5) { printf qq{%s_%d }, $n, (5000000 * $i); } }');
do
    DIR_COUNT="${BASE_DIR}/${d}/"
    
    if [ ! -d ${DIR_COUNT} ]; then
        continue     
    fi
    
    bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 1 ${DIR_COUNT} ${REAL_G} \
        >> ${BASE_DIR}/stat1.md
done

cat stat1.md
```

* Stats of anchors

```bash
BASE_DIR=$HOME/data/anchr/n2
cd ${BASE_DIR}

bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 2 header \
    > ${BASE_DIR}/stat2.md

for d in $(perl -e 'for $n (qw{Q20L80}) { for $i (1 .. 5) { printf qq{%s_%d }, $n, (5000000 * $i); } }');
do
    DIR_COUNT="${BASE_DIR}/${d}/"
    
    if [ ! -e ${DIR_COUNT}/anchor/pe.anchor.fa ]; then
        continue     
    fi
    
    bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 2 ${DIR_COUNT} \
        >> ${BASE_DIR}/stat2.md

done

cat stat2.md
```

| Name            |   SumFq | CovFq | AvgRead | Kmer |   SumFa | Discard% |   RealG |   EstG | Est/Real |   SumKU | SumSR |   RunTime |
|:----------------|--------:|------:|--------:|-----:|--------:|---------:|--------:|-------:|---------:|--------:|------:|----------:|
| Q20L80_5000000  | 980.18M |   9.8 |      97 |   71 | 897.53M |   8.432% | 100.29M | 87.08M |     0.87 | 117.08M |     0 | 0:05'56'' |
| Q20L80_10000000 |   1.96G |  19.5 |      97 |   71 |   1.82G |   7.346% | 100.29M | 95.68M |     0.95 | 118.27M |     0 | 0:11'08'' |
| Q20L80_15000000 |   2.94G |  29.3 |      97 |   71 |   2.73G |   7.223% | 100.29M | 97.32M |     0.97 | 115.98M |     0 | 0:16'45'' |
| Q20L80_20000000 |   3.92G |  39.1 |      97 |   71 |   3.64G |   7.175% | 100.29M | 97.93M |     0.98 | 115.25M |     0 | 0:19'59'' |
| Q20L80_25000000 |   4.76G |  47.5 |      97 |   71 |   4.42G |   7.153% | 100.29M | 98.21M |     0.98 |  115.3M |     0 | 0:26'45'' |

| Name            | N50SRclean |    Sum |     # | N50Anchor |    Sum |     # | N50Anchor2 |   Sum | # | N50Others |    Sum |     # |   RunTime |
|:----------------|-----------:|-------:|------:|----------:|-------:|------:|-----------:|------:|--:|----------:|-------:|------:|----------:|
| Q20L80_5000000  |        637 | 12.58M | 19037 |      1190 |  1.05M |   845 |          0 |     0 | 0 |       621 | 11.52M | 18192 | 0:04'29'' |
| Q20L80_10000000 |       1105 | 64.74M | 62975 |      1604 | 36.11M | 22378 |       1096 | 2.16K | 2 |       725 | 28.63M | 40595 | 0:09'23'' |
| Q20L80_15000000 |       1994 | 83.29M | 53022 |      2499 | 66.13M | 29259 |       2455 | 7.13K | 4 |       749 | 17.15M | 23759 | 0:13'53'' |
| Q20L80_20000000 |       3050 | 89.24M | 42590 |      3564 | 77.94M | 26959 |       1750 | 3.25K | 2 |       752 |  11.3M | 15629 | 0:17'21'' |
| Q20L80_25000000 |       3831 | 91.59M | 37420 |      4373 | 82.57M | 24972 |       1890 | 1.89K | 1 |       754 |  9.02M | 12447 | 0:20'04'' |

## Cele: quality assessment

```bash
BASE_DIR=$HOME/data/anchr/n2
cd ${BASE_DIR}

for part in anchor anchor2 others;
do 
    bash ~/Scripts/cpan/App-Anchr/share/sort_on_ref.sh Q20L80_25000000/anchor/pe.${part}.fa 1_genome/genome.fa pe.${part}
    nucmer -l 200 1_genome/genome.fa pe.${part}.fa
    mummerplot -png out.delta -p pe.${part} --medium
done

# mummerplot files
rm *.[fr]plot
rm out.delta
rm *.gp

cp ~/data/anchr/paralogs/model/Results/n2/n2.multi.fas 1_genome/paralogs.fas

# quast
rm -fr 9_qa
quast --no-check \
    -R 1_genome/genome.fa \
    Q20L80_25000000/anchor/pe.anchor.fa \
    1_genome/paralogs.fas \
    --label "25000000,paralogs" \
    -o 9_qa
```

# *Arabidopsis thaliana* Col-0

* Genome: [Ensembl Genomes](http://plants.ensembl.org/Arabidopsis_thaliana/Info/Index)
* Proportion of paralogs (> 1000 bp): 0.1158

## Atha: download

* Reference genome

```bash
mkdir -p ~/data/anchr/col_0/1_genome
cd ~/data/anchr/col_0/1_genome
wget -N ftp://ftp.ensemblgenomes.org/pub/release-29/plants/fasta/arabidopsis_thaliana/dna/Arabidopsis_thaliana.TAIR10.29.dna_sm.toplevel.fa.gz
faops order Arabidopsis_thaliana.TAIR10.29.dna_sm.toplevel.fa.gz \
    <(for chr in {1,2,3,4,5,Mt,Pt}; do echo $chr; done) \
    genome.fa
```

* Illumina

```bash
# Downloading from ena with aria2
mkdir -p ~/data/anchr/col_0/2_illumina
cd ~/data/anchr/col_0/2_illumina
aria2c -x 9 -s 3 -c ftp://ftp.sra.ebi.ac.uk/vol1/srr/SRR611/SRR611086
fastq-dump --split-files ./SRR611086
find . -name "*.fastq" | parallel -j 2 pigz -p 8

ln -s SRR611086_1.fastq.gz R1.fq.gz
ln -s SRR611086_2.fastq.gz R2.fq.gz
```

* PacBio

```bash
mkdir -p ~/data/anchr/col_0/3_pacbio

```

## Atha: trim

* Q20L80

```bash
mkdir -p ~/data/anchr/col_0/2_illumina/Q20L80
pushd ~/data/anchr/col_0/2_illumina/Q20L80

anchr trim \
    -q 20 -l 80 \
    ../R1.fq.gz ../R2.fq.gz \
    -o stdout \
    | bash

popd
```

* Stats

```bash
cd ~/data/anchr/col_0

printf "| %s | %s | %s | %s |\n" \
    "Name" "N50" "Sum" "#" \
    > stat.md
printf "|:--|--:|--:|--:|\n" >> stat.md

printf "| %s | %s | %s | %s |\n" \
    $(echo "Genome";   faops n50 -H -S -C 1_genome/genome.fa;) >> stat.md
printf "| %s | %s | %s | %s |\n" \
    $(echo "Illumina"; faops n50 -H -S -C 2_illumina/R1.fq.gz 2_illumina/R2.fq.gz;) >> stat.md
printf "| %s | %s | %s | %s |\n" \
    $(echo "PacBio";   faops n50 -H -S -C 3_pacbio/pacbio.fasta;) >> stat.md

printf "| %s | %s | %s | %s |\n" \
    $(echo "Q20L80";  faops n50 -H -S -C 2_illumina/Q20L80/R1.fq.gz 2_illumina/Q20L80/R1.fq.gz;) >> stat.md

cat stat.md
```

| Name     |      N50 |        Sum |        # |
|:---------|---------:|-----------:|---------:|
| Genome   | 23459830 |  119667750 |        7 |
| Illumina |      100 | 9978269800 | 99782698 |
| PacBio   |          |            |          |
| Q20L80   |      100 | 8618615500 | 86472520 |

## Atha: down sampling

```bash
BASE_DIR=$HOME/data/anchr/col_0
cd ${BASE_DIR}

# works on bash 3
ARRAY=( "2_illumina/Q20L80:Q20L80:40000000")

for group in "${ARRAY[@]}" ; do
    
    GROUP_DIR=$(group=${group} perl -e '@p = split q{:}, $ENV{group}; print $p[0];')
    GROUP_ID=$( group=${group} perl -e '@p = split q{:}, $ENV{group}; print $p[1];')
    GROUP_MAX=$(group=${group} perl -e '@p = split q{:}, $ENV{group}; print $p[2];')
    printf "==> %s \t %s \t %s\n" "$GROUP_DIR" "$GROUP_ID" "$GROUP_MAX"

    for count in $(perl -e 'print 5000000 * $_, q{ } for 1 .. 8');
    do
        if [[ "$count" -gt "$GROUP_MAX" ]]; then
            continue     
        fi
        
        echo "==> Reads ${GROUP_ID}_${count}"
        DIR_COUNT="${BASE_DIR}/${GROUP_ID}_${count}"
        mkdir -p ${DIR_COUNT}
        
        if [ -e ${DIR_COUNT}/R1.fq.gz ]; then
            continue     
        fi
        
        seqtk sample -s${count} \
            ${BASE_DIR}/${GROUP_DIR}/R1.fq.gz ${count} \
            | pigz > ${DIR_COUNT}/R1.fq.gz
        seqtk sample -s${count} \
            ${BASE_DIR}/${GROUP_DIR}/R2.fq.gz ${count} \
            | pigz > ${DIR_COUNT}/R2.fq.gz
    done
done
```

## Atha: generate super-reads

```bash
BASE_DIR=$HOME/data/anchr/col_0
cd ${BASE_DIR}

for d in $(perl -e 'for $n (qw{Q20L80}) { for $i (1 .. 8) { printf qq{%s_%d }, $n, (5000000 * $i); } }');
do
    echo
    echo "==> Reads ${d}"
    DIR_COUNT="${BASE_DIR}/${d}/"

    if [ ! -d ${DIR_COUNT} ]; then
        continue
    fi
    
    if [ -e ${DIR_COUNT}/pe.cor.fa ]; then
        echo "    pe.cor.fa already presents"
        continue
    fi
    
    pushd ${DIR_COUNT} > /dev/null
    anchr superreads \
        R1.fq.gz \
        R2.fq.gz \
        --nosr \
        -s 200 -d 20 -p 16
    bash superreads.sh
    popd > /dev/null
done
```

Clear intermediate files.

```bash
cd $HOME/data/anchr/col_0/

find . -type f -name "quorum_mer_db.jf" | xargs rm
find . -type f -name "k_u_hash_0" | xargs rm
find . -type f -name "readPositionsInSuperReads" | xargs rm
find . -type f -name "*.tmp" | xargs rm
#find . -type f -name "pe.renamed.fastq" | xargs rm
```

## Atha: create anchors

```bash
BASE_DIR=$HOME/data/anchr/col_0
cd ${BASE_DIR}

for d in $(perl -e 'for $n (qw{Q20L80}) { for $i (1 .. 8) { printf qq{%s_%d }, $n, (5000000 * $i); } }');
do
    echo
    echo "==> Reads ${d}"
    DIR_COUNT="${BASE_DIR}/${d}/"
    
    if [ -e ${DIR_COUNT}/anchor/pe.anchor.fa ]; then
        continue
    fi
    
    rm -fr ${DIR_COUNT}/anchor
    bash ~/Scripts/cpan/App-Anchr/share/anchor.sh ${DIR_COUNT} 16 80 false
done
```

## Atha: results

* Stats of super-reads

```bash
BASE_DIR=$HOME/data/anchr/col_0
cd ${BASE_DIR}

REAL_G=119667750

bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 1 header \
    > ${BASE_DIR}/stat1.md

for d in $(perl -e 'for $n (qw{Q20L80}) { for $i (1 .. 8) { printf qq{%s_%d }, $n, (5000000 * $i); } }');
do
    DIR_COUNT="${BASE_DIR}/${d}/"
    
    if [ ! -d ${DIR_COUNT} ]; then
        continue     
    fi
    
    bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 1 ${DIR_COUNT} ${REAL_G} \
        >> ${BASE_DIR}/stat1.md
done

cat stat1.md
```

* Stats of anchors

```bash
BASE_DIR=$HOME/data/anchr/col_0
cd ${BASE_DIR}

bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 2 header \
    > ${BASE_DIR}/stat2.md

for d in $(perl -e 'for $n (qw{Q20L80}) { for $i (1 .. 8) { printf qq{%s_%d }, $n, (5000000 * $i); } }');
do
    DIR_COUNT="${BASE_DIR}/${d}/"
    
    if [ ! -e ${DIR_COUNT}/anchor/pe.anchor.fa ]; then
        continue     
    fi
    
    bash ~/Scripts/cpan/App-Anchr/share/sr_stat.sh 2 ${DIR_COUNT} \
        >> ${BASE_DIR}/stat2.md

done

cat stat2.md
```

| Name            |   SumFq | CovFq | AvgRead | Kmer |   SumFa | Discard% |   RealG |    EstG | Est/Real |   SumKU | SumSR |   RunTime |
|:----------------|--------:|------:|--------:|-----:|--------:|---------:|--------:|--------:|---------:|--------:|------:|----------:|
| Q20L80_5000000  | 994.55M |   8.3 |      99 |   71 | 707.85M |  28.828% | 119.67M |  80.47M |     0.67 | 115.95M |     0 | 0:05'57'' |
| Q20L80_10000000 |   1.99G |  16.6 |      99 |   71 |   1.51G |  24.124% | 119.67M | 114.21M |     0.95 | 160.41M |     0 | 0:13'30'' |
| Q20L80_15000000 |   2.98G |  24.9 |      99 |   71 |    2.3G |  22.808% | 119.67M |  126.9M |     1.06 | 169.83M |     0 | 0:17'29'' |
| Q20L80_20000000 |   3.98G |  33.2 |      99 |   71 |   3.11G |  21.758% | 119.67M | 139.64M |     1.17 | 182.61M |     0 | 0:21'04'' |
| Q20L80_25000000 |   4.97G |  41.6 |      99 |   71 |   3.94G |  20.820% | 119.67M | 154.03M |     1.29 | 201.12M |     0 | 0:33'45'' |
| Q20L80_30000000 |   5.97G |  49.9 |      99 |   71 |   4.77G |  20.007% | 119.67M | 169.58M |     1.42 |  223.3M |     0 | 0:25'15'' |
| Q20L80_35000000 |   6.96G |  58.2 |      99 |   71 |   5.62G |  19.289% | 119.67M | 185.76M |     1.55 | 247.36M |     0 | 0:30'35'' |
| Q20L80_40000000 |   7.96G |  66.5 |      99 |   71 |   6.47G |  18.671% | 119.67M | 202.11M |     1.69 | 272.18M |     0 | 0:37'11'' |

| Name            | N50SRclean |     Sum |     # | N50Anchor |     Sum |     # | N50Anchor2 |   Sum | # | N50Others |     Sum |     # |   RunTime |
|:----------------|-----------:|--------:|------:|----------:|--------:|------:|-----------:|------:|--:|----------:|--------:|------:|----------:|
| Q20L80_5000000  |       1500 | 532.36K |   448 |         0 |       0 |     0 |          0 |     0 | 0 |      1500 | 532.36K |   448 | 0:02'45'' |
| Q20L80_10000000 |        656 |   14.5M | 21372 |         0 |       0 |     0 |          0 |     0 | 0 |       656 |   14.5M | 21372 | 0:04'53'' |
| Q20L80_15000000 |        941 |  62.58M | 69252 |      1387 |  27.67M | 19474 |          0 |     0 | 0 |       717 |   34.9M | 49778 | 0:09'27'' |
| Q20L80_20000000 |       1639 |  92.22M | 67020 |      2023 |  70.15M | 36628 |       1086 | 3.41K | 3 |       756 |  22.07M | 30389 | 0:15'54'' |
| Q20L80_25000000 |       3005 | 102.07M | 47207 |      3327 |  91.66M | 33019 |          0 |     0 | 0 |       771 |  10.41M | 14188 | 0:15'54'' |
| Q20L80_30000000 |       4902 | 105.17M | 33963 |      5217 |  99.16M | 25721 |       1776 | 3.23K | 2 |       766 |   6.01M |  8240 | 0:18'46'' |
| Q20L80_35000000 |       6833 | 106.59M | 27391 |      7164 | 101.99M | 21035 |       1089 | 1.09K | 1 |       756 |    4.6M |  6355 | 0:20'51'' |
| Q20L80_40000000 |       8220 |  107.4M | 24504 |      8611 | 103.23M | 18700 |          0 |     0 | 0 |       748 |   4.17M |  5804 | 0:24'19'' |

## Atha: quality assessment

```bash
BASE_DIR=$HOME/data/anchr/col_0
cd ${BASE_DIR}

for part in anchor anchor2 others;
do 
    bash ~/Scripts/cpan/App-Anchr/share/sort_on_ref.sh Q20L80_40000000/anchor/pe.${part}.fa 1_genome/genome.fa pe.${part}
    nucmer -l 200 1_genome/genome.fa pe.${part}.fa
    mummerplot -png out.delta -p pe.${part} --medium
done

# mummerplot files
rm *.[fr]plot
rm out.delta
rm *.gp

cp ~/data/anchr/paralogs/model/Results/col_0/col_0.multi.fas 1_genome/paralogs.fas

# quast
rm -fr 9_qa
quast --no-check \
    -R 1_genome/genome.fa \
    Q20L80_30000000/anchor/pe.anchor.fa \
    Q20L80_40000000/anchor/pe.anchor.fa \
    1_genome/paralogs.fas \
    --label "30000000,40000000,paralogs" \
    -o 9_qa
```
